{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Brendan\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3063: DtypeWarning: Columns (4,6,8,9,10) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n",
      "C:\\Users\\Brendan\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3063: DtypeWarning: Columns (7) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import re\n",
    "import os\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "refseq - ncbi-ftp/gene/DATA/gene2refseq.gz\n",
    "pubmed - ncbi-ftp/gene/DATA/gene2pubmed.gz\n",
    "ensembl - ncbi-ftp/gene/DATA/gene2ensembl.gz\n",
    "human - ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/Mammalia/Homo_sapiens.gene_info.gz\n",
    "mart - useast.ensembl.org/index.html\n",
    "neighbors - ncbi-ftp/gene/DATA/gene_neighbors.gz\n",
    "orthologs - ncbi-ftp/gene/DATA/gene_orthologs.gz\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "refseq = pd.read_table('gene2refseq')\n",
    "pubmed = pd.read_table('gene2pubmed')\n",
    "ensembl = pd.read_table('gene2ensembl')\n",
    "mart = pd.read_csv('mart_export.txt',delimiter = ',')\n",
    "neighbors = pd.read_table('gene_neighbors')\n",
    "orthologs = pd.read_table('gene_orthologs')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#tax_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>status</th>\n",
       "      <th>RNA_nucleotide_accession.version</th>\n",
       "      <th>RNA_nucleotide_gi</th>\n",
       "      <th>protein_accession.version</th>\n",
       "      <th>protein_gi</th>\n",
       "      <th>genomic_nucleotide_accession.version</th>\n",
       "      <th>genomic_nucleotide_gi</th>\n",
       "      <th>start_position_on_the_genomic_accession</th>\n",
       "      <th>end_position_on_the_genomic_accession</th>\n",
       "      <th>orientation</th>\n",
       "      <th>assembly</th>\n",
       "      <th>mature_peptide_accession.version</th>\n",
       "      <th>mature_peptide_gi</th>\n",
       "      <th>Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>1246500</td>\n",
       "      <td>PROVISIONAL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NP_047184.1</td>\n",
       "      <td>10954455</td>\n",
       "      <td>NC_001911.1</td>\n",
       "      <td>10954454</td>\n",
       "      <td>348</td>\n",
       "      <td>1190</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>repA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>1246501</td>\n",
       "      <td>PROVISIONAL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NP_047186.1</td>\n",
       "      <td>10954457</td>\n",
       "      <td>NC_001911.1</td>\n",
       "      <td>10954454</td>\n",
       "      <td>2157</td>\n",
       "      <td>2912</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>repA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>1246502</td>\n",
       "      <td>VALIDATED</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NP_047187.1</td>\n",
       "      <td>10954458</td>\n",
       "      <td>NC_001911.1</td>\n",
       "      <td>10954454</td>\n",
       "      <td>3040</td>\n",
       "      <td>4590</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>leuA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>1246503</td>\n",
       "      <td>PROVISIONAL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NP_047188.1</td>\n",
       "      <td>10954459</td>\n",
       "      <td>NC_001911.1</td>\n",
       "      <td>10954454</td>\n",
       "      <td>4623</td>\n",
       "      <td>5714</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>leuB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>1246504</td>\n",
       "      <td>VALIDATED</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NP_047189.1</td>\n",
       "      <td>10954460</td>\n",
       "      <td>NC_001911.1</td>\n",
       "      <td>10954454</td>\n",
       "      <td>5717</td>\n",
       "      <td>7117</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>leuC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40141439</th>\n",
       "      <td>2749911</td>\n",
       "      <td>36959328</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NC_038002.1</td>\n",
       "      <td>1418587958</td>\n",
       "      <td>119755</td>\n",
       "      <td>119828</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>trnP-GGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40141440</th>\n",
       "      <td>2749911</td>\n",
       "      <td>36959329</td>\n",
       "      <td>PROVISIONAL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>YP_009496746.1</td>\n",
       "      <td>1418587959</td>\n",
       "      <td>NC_038002.1</td>\n",
       "      <td>1418587958</td>\n",
       "      <td>116</td>\n",
       "      <td>2374</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>psaA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40141441</th>\n",
       "      <td>2749911</td>\n",
       "      <td>36959330</td>\n",
       "      <td>PROVISIONAL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>YP_009496877.1</td>\n",
       "      <td>1418588090</td>\n",
       "      <td>NC_038002.1</td>\n",
       "      <td>1418587958</td>\n",
       "      <td>120098</td>\n",
       "      <td>120337</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>acpP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40141442</th>\n",
       "      <td>2749911</td>\n",
       "      <td>36959331</td>\n",
       "      <td>PROVISIONAL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>YP_009496787.1</td>\n",
       "      <td>1418588000</td>\n",
       "      <td>NC_038002.1</td>\n",
       "      <td>1418587958</td>\n",
       "      <td>40018</td>\n",
       "      <td>40134</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>psbI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40141443</th>\n",
       "      <td>2749911</td>\n",
       "      <td>36959332</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NC_038002.1</td>\n",
       "      <td>1418587958</td>\n",
       "      <td>67821</td>\n",
       "      <td>69298</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rns</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>40141444 rows Ã— 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          #tax_id    GeneID       status RNA_nucleotide_accession.version  \\\n",
       "0               9   1246500  PROVISIONAL                                -   \n",
       "1               9   1246501  PROVISIONAL                                -   \n",
       "2               9   1246502    VALIDATED                                -   \n",
       "3               9   1246503  PROVISIONAL                                -   \n",
       "4               9   1246504    VALIDATED                                -   \n",
       "...           ...       ...          ...                              ...   \n",
       "40141439  2749911  36959328          NaN                                -   \n",
       "40141440  2749911  36959329  PROVISIONAL                                -   \n",
       "40141441  2749911  36959330  PROVISIONAL                                -   \n",
       "40141442  2749911  36959331  PROVISIONAL                                -   \n",
       "40141443  2749911  36959332          NaN                                -   \n",
       "\n",
       "         RNA_nucleotide_gi protein_accession.version  protein_gi  \\\n",
       "0                        -               NP_047184.1    10954455   \n",
       "1                        -               NP_047186.1    10954457   \n",
       "2                        -               NP_047187.1    10954458   \n",
       "3                        -               NP_047188.1    10954459   \n",
       "4                        -               NP_047189.1    10954460   \n",
       "...                    ...                       ...         ...   \n",
       "40141439                 -                         -           -   \n",
       "40141440                 -            YP_009496746.1  1418587959   \n",
       "40141441                 -            YP_009496877.1  1418588090   \n",
       "40141442                 -            YP_009496787.1  1418588000   \n",
       "40141443                 -                         -           -   \n",
       "\n",
       "         genomic_nucleotide_accession.version genomic_nucleotide_gi  \\\n",
       "0                                 NC_001911.1              10954454   \n",
       "1                                 NC_001911.1              10954454   \n",
       "2                                 NC_001911.1              10954454   \n",
       "3                                 NC_001911.1              10954454   \n",
       "4                                 NC_001911.1              10954454   \n",
       "...                                       ...                   ...   \n",
       "40141439                          NC_038002.1            1418587958   \n",
       "40141440                          NC_038002.1            1418587958   \n",
       "40141441                          NC_038002.1            1418587958   \n",
       "40141442                          NC_038002.1            1418587958   \n",
       "40141443                          NC_038002.1            1418587958   \n",
       "\n",
       "         start_position_on_the_genomic_accession  \\\n",
       "0                                            348   \n",
       "1                                           2157   \n",
       "2                                           3040   \n",
       "3                                           4623   \n",
       "4                                           5717   \n",
       "...                                          ...   \n",
       "40141439                                  119755   \n",
       "40141440                                     116   \n",
       "40141441                                  120098   \n",
       "40141442                                   40018   \n",
       "40141443                                   67821   \n",
       "\n",
       "         end_position_on_the_genomic_accession orientation assembly  \\\n",
       "0                                         1190           -        -   \n",
       "1                                         2912           +        -   \n",
       "2                                         4590           +        -   \n",
       "3                                         5714           +        -   \n",
       "4                                         7117           +        -   \n",
       "...                                        ...         ...      ...   \n",
       "40141439                                119828           -        -   \n",
       "40141440                                  2374           +        -   \n",
       "40141441                                120337           -        -   \n",
       "40141442                                 40134           +        -   \n",
       "40141443                                 69298           +        -   \n",
       "\n",
       "         mature_peptide_accession.version mature_peptide_gi    Symbol  \n",
       "0                                       -                 -     repA1  \n",
       "1                                       -                 -     repA2  \n",
       "2                                       -                 -      leuA  \n",
       "3                                       -                 -      leuB  \n",
       "4                                       -                 -      leuC  \n",
       "...                                   ...               ...       ...  \n",
       "40141439                                -                 -  trnP-GGG  \n",
       "40141440                                -                 -      psaA  \n",
       "40141441                                -                 -      acpP  \n",
       "40141442                                -                 -      psbI  \n",
       "40141443                                -                 -       rns  \n",
       "\n",
       "[40141444 rows x 16 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human = pd.read_table('Homo_sapiens.gene_info')\n",
    "aliases = human['Synonyms'].tolist()\n",
    "aliases_sep = []\n",
    "for i in range(len(aliases)):\n",
    "    my_list = aliases[i].split(\"|\")\n",
    "    my_list.append(human['Symbol'].iloc[i])\n",
    "    aliases_sep.append(my_list)\n",
    "human['Aliases'] = aliases_sep\n",
    "refseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def compare_sets(list_data):\n",
    "    for ele in list_data:\n",
    "        col_names = ele.columns\n",
    "        print(col_names)\n",
    "data_list = refseq,pubmed,ensembl,human,mart,neighbors,orthologs\n",
    "# compare_sets(data_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "def combined_translate(input_file, file_type, trans_list):\n",
    "    if (file_type == 'csv'):\n",
    "        df = pd.read_csv(input_file)\n",
    "    if (file_type == 'excel'):\n",
    "        df = pd.read_excel(input_file)\n",
    "    if (file_type == 'other'):\n",
    "        df = pd.read_table(input_file)\n",
    "    \n",
    "    for item in trans_list:\n",
    "        test_col = item[0]\n",
    "        reference_df = item[1]\n",
    "        reference_col = item[2]\n",
    "        output_col = item[3]\n",
    "        \n",
    "        \n",
    "        start_param = df[test_col].tolist()\n",
    "        in_list = reference_df[reference_col].tolist()\n",
    "        out_listoflist = []\n",
    "\n",
    "        for i in range(len(start_param)):\n",
    "            param = start_param[i]\n",
    "\n",
    "            match_list=[]\n",
    "            for j in range(len(in_list)):\n",
    "                if str(param) in str(in_list[j]):\n",
    "                    index_match = reference_df.iloc[j]\n",
    "                    match_list.append(index_match)\n",
    "\n",
    "            out_list = []\n",
    "            for k in range(len(match_list)):\n",
    "                if (output_col == 'MATCH'):\n",
    "                    out_list.append(match_list[k][reference_col])\n",
    "                else: \n",
    "                    out_list.append(match_list[k][output_col])\n",
    "\n",
    "            out_listoflist.append(out_list)\n",
    "        out_colname = 'output ' + reference_col\n",
    "        df[out_colname] = out_listoflist\n",
    "\n",
    "    \n",
    "        nonmatches_ind = [ind for ind, x in enumerate(df[out_colname]) if len(x)==0 or x != x or x=='-']\n",
    "        matches_ind = [ind for ind, x in enumerate(df[out_colname]) if len(x)!=0 or x == x or x!='-']\n",
    "        nonmatches = df.iloc[nonmatches_ind]\n",
    "        matches = df.iloc[matches_ind]\n",
    "        \n",
    "        return str(round(100*len(nonmatches)/len(df),4))+' % mismatch '\n",
    "#         print(round(100*len(nonmatches)/len(df),4),\" percent mismatch, \",input_file, 'to')\n",
    "    \n",
    "#     print(df)\n",
    "    \n",
    "        \n",
    "          "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13.099 % mismatch md_2018_05_18_zhu_md-d-17-05991_sdc1.xlsx to Aliases\n",
      "12.7796 % mismatch md_2018_05_18_zhu_md-d-17-05991_sdc1.xlsx to GeneID\n",
      "7.5472 % mismatch RA_loci.txt to Aliases\n",
      "10.4762 % mismatch Yamamoto.txt to Aliases\n",
      "19.0476 % mismatch RADB.txt to Aliases\n"
     ]
    }
   ],
   "source": [
    "### Name of input data file\n",
    "input_file = 'md_2018_05_18_zhu_md-d-17-05991_sdc1.xlsx'\n",
    "\n",
    "### Type of file (excel, csv, or other)\n",
    "file_type = 'excel'\n",
    "\n",
    "### translation 1 test col name\n",
    "test_col1 = 'Gene Symbol'\n",
    "\n",
    "### translation 1 reference dataset (human,gene,mart)\n",
    "ref_df1 = human\n",
    "\n",
    "### translation 1 reference col name\n",
    "ref_col1 = 'Aliases'\n",
    "    \n",
    "### translation 1 output col name\n",
    "out_col1 = 'MATCH'\n",
    "\n",
    "### translation 1 use only if one translation, otherwise delete\n",
    "trans_list = [[test_col1,ref_df1,ref_col1,out_col1]]\n",
    "print(combined_translate(input_file,file_type,trans_list))\n",
    "\n",
    "\n",
    "\n",
    "input_file = 'md_2018_05_18_zhu_md-d-17-05991_sdc1.xlsx'\n",
    "file_type = 'excel'\n",
    "test_col1 = 'Entrez Gene'\n",
    "ref_df1 = human\n",
    "ref_col1 = 'GeneID'\n",
    "out_col1 = 'MATCH'\n",
    "trans_list = [[test_col1,ref_df1,ref_col1,out_col1]]\n",
    "print(combined_translate(input_file,file_type,trans_list))\n",
    "\n",
    "\n",
    "\n",
    "input_file = 'RA_loci.txt'\n",
    "file_type = 'csv'\n",
    "test_col1 = 'Gene'\n",
    "ref_df1 = human\n",
    "ref_col1 = 'Aliases'\n",
    "out_col1 = 'MATCH'\n",
    "trans_list = [[test_col1,ref_df1,ref_col1,out_col1]]\n",
    "print(combined_translate(input_file,file_type,trans_list))\n",
    "\n",
    "\n",
    "\n",
    "input_file = 'Yamamoto.txt'\n",
    "file_type = 'other'\n",
    "test_col1 = 'Gene'\n",
    "ref_df1 = human\n",
    "ref_col1 = 'Aliases'\n",
    "out_col1 = 'MATCH'\n",
    "trans_list = [[test_col1,ref_df1,ref_col1,out_col1]]\n",
    "print(combined_translate(input_file,file_type,trans_list))\n",
    "\n",
    "\n",
    "\n",
    "input_file = 'RADB.txt'\n",
    "file_type = 'other'\n",
    "test_col1 = 'Gene'\n",
    "ref_df1 = human\n",
    "ref_col1 = 'Aliases'\n",
    "out_col1 = 'MATCH'\n",
    "trans_list = [[test_col1,ref_df1,ref_col1,out_col1]]\n",
    "print(combined_translate(input_file,file_type,trans_list))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "EOL while scanning string literal (<ipython-input-98-f263c23531b2>, line 24)",
     "output_type": "error",
     "traceback": [
      "\u001b[1;36m  File \u001b[1;32m\"<ipython-input-98-f263c23531b2>\"\u001b[1;36m, line \u001b[1;32m24\u001b[0m\n\u001b[1;33m    print(match_percent,\" data_list[i][0], 'to', data_list[j][0])\u001b[0m\n\u001b[1;37m                                                                 ^\u001b[0m\n\u001b[1;31mSyntaxError\u001b[0m\u001b[1;31m:\u001b[0m EOL while scanning string literal\n"
     ]
    }
   ],
   "source": [
    "def match_lists(data_list):\n",
    "    for i in range(len(data_list)):\n",
    "        for j in range(len(data_list)):\n",
    "            if (i != j):\n",
    "                ### translation 1 test col name\n",
    "                test_col1 = data_list[i][1]\n",
    "\n",
    "                ### translation 1 reference dataset (human,gene,mart)\n",
    "                if (data_list[j][2] == 'csv'):\n",
    "                    ref_df1 = pd.read_csv(data_list[j][0])\n",
    "                if (data_list[j][2] == 'excel'):\n",
    "                    ref_df1 = pd.read_excel(data_list[j][0])\n",
    "                if (data_list[j][2] == 'other'):\n",
    "                    ref_df1 = pd.read_table(data_list[j][0])\n",
    "\n",
    "                ### translation 1 reference col name\n",
    "                ref_col1 = data_list[j][1]\n",
    "\n",
    "                ### translation 1 output col name\n",
    "                out_col1 = 'MATCH'\n",
    "                \n",
    "                trans_list = [[test_col1,ref_df1,ref_col1,out_col1]]\n",
    "                match_percent = combined_translate(data_list[i][0], data_list[i][2], trans_list)\n",
    "                print(match_percent,\" data_list[i][0], 'to', data_list[j][0])\n",
    "#                 print(match_percent)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "96.8051 % mismatch   percent mismatch,  md_2018.xlsx to ra_loci.txt\n",
      "96.8051 % mismatch   percent mismatch,  md_2018.xlsx to Yamamoto.txt\n",
      "99.0415 % mismatch   percent mismatch,  md_2018.xlsx to RADB.txt\n",
      "96.8051 % mismatch   percent mismatch,  md_2018.xlsx to okada.txt\n",
      "90.566 % mismatch   percent mismatch,  ra_loci.txt to md_2018.xlsx\n",
      "15.0943 % mismatch   percent mismatch,  ra_loci.txt to Yamamoto.txt\n",
      "76.4151 % mismatch   percent mismatch,  ra_loci.txt to RADB.txt\n",
      "14.1509 % mismatch   percent mismatch,  ra_loci.txt to okada.txt\n",
      "92.381 % mismatch   percent mismatch,  Yamamoto.txt to md_2018.xlsx\n",
      "19.0476 % mismatch   percent mismatch,  Yamamoto.txt to ra_loci.txt\n",
      "80.9524 % mismatch   percent mismatch,  Yamamoto.txt to RADB.txt\n",
      "9.5238 % mismatch   percent mismatch,  Yamamoto.txt to okada.txt\n",
      "95.2381 % mismatch   percent mismatch,  RADB.txt to md_2018.xlsx\n",
      "19.0476 % mismatch   percent mismatch,  RADB.txt to ra_loci.txt\n",
      "14.2857 % mismatch   percent mismatch,  RADB.txt to Yamamoto.txt\n",
      "14.2857 % mismatch   percent mismatch,  RADB.txt to okada.txt\n",
      "92.0 % mismatch   percent mismatch,  okada.txt to md_2018.xlsx\n",
      "23.0 % mismatch   percent mismatch,  okada.txt to ra_loci.txt\n",
      "14.0 % mismatch   percent mismatch,  okada.txt to Yamamoto.txt\n",
      "80.0 % mismatch   percent mismatch,  okada.txt to RADB.txt\n"
     ]
    }
   ],
   "source": [
    "md_2018 = pd.read_excel('md_2018.xlsx')\n",
    "# https://journals.lww.com/md-journal/Fulltext/2018/06010/Identification_of_key_genes_in_rheumatoid.86.aspx\n",
    "# A total of 313 genes (DEGs) were identified to be differentially expressed between RA and NC samples\n",
    "\n",
    "ra_loci = pd.read_csv('RA_loci.txt')\n",
    "# https://academic.oup.com/view-large/27924154\n",
    "# Genetic loci associated with susceptibility to RA\n",
    "\n",
    "yamamoto = pd.read_table('Yamamoto.txt')\n",
    "# https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4729856/\n",
    "# RA susceptible genes\n",
    "\n",
    "radb = pd.read_table('RADB.txt')\n",
    "# https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4164886/\n",
    "# Genes and genetic regions that have the strongest association with RA susceptibility. \n",
    "\n",
    "okada = pd.read_table('okada.txt')\n",
    "# https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3944098/#SD2\n",
    "# Novel rheumatoid arthritis risk loci identified by trans-ethnic GWAS meta-analysis in >100,000 subjects.\n",
    "\n",
    "\n",
    "l1 = ['md_2018.xlsx', 'Gene Symbol','excel']\n",
    "l2 = ['ra_loci.txt','Gene','csv']\n",
    "l3 = ['Yamamoto.txt','Gene','other']\n",
    "l4 = ['RADB.txt','Gene','other']\n",
    "l5 = ['okada.txt','Gene','other']\n",
    "\n",
    "match_lists([l1,l2,l3,l4,l5])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
