{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n\\nrefseq - ncbi-ftp/gene/DATA/gene2refseq.gz\\npubmed - ncbi-ftp/gene/DATA/gene2pubmed.gz\\nensembl - ncbi-ftp/gene/DATA/gene2ensembl.gz\\nhuman - ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/Mammalia/Homo_sapiens.gene_info.gz\\nmart - useast.ensembl.org/index.html\\nneighbors - ncbi-ftp/gene/DATA/gene_neighbors.gz\\northologs - ncbi-ftp/gene/DATA/gene_orthologs.gz\\n\\n'"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import re\n",
    "import os\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "refseq - ncbi-ftp/gene/DATA/gene2refseq.gz\n",
    "pubmed - ncbi-ftp/gene/DATA/gene2pubmed.gz\n",
    "ensembl - ncbi-ftp/gene/DATA/gene2ensembl.gz\n",
    "human - ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/GENE_INFO/Mammalia/Homo_sapiens.gene_info.gz\n",
    "mart - useast.ensembl.org/index.html\n",
    "neighbors - ncbi-ftp/gene/DATA/gene_neighbors.gz\n",
    "orthologs - ncbi-ftp/gene/DATA/gene_orthologs.gz\n",
    "\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Brendan\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3063: DtypeWarning: Columns (4,6,8,9,10) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#tax_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>status</th>\n",
       "      <th>RNA_nucleotide_accession.version</th>\n",
       "      <th>RNA_nucleotide_gi</th>\n",
       "      <th>protein_accession.version</th>\n",
       "      <th>protein_gi</th>\n",
       "      <th>genomic_nucleotide_accession.version</th>\n",
       "      <th>genomic_nucleotide_gi</th>\n",
       "      <th>start_position_on_the_genomic_accession</th>\n",
       "      <th>end_position_on_the_genomic_accession</th>\n",
       "      <th>orientation</th>\n",
       "      <th>assembly</th>\n",
       "      <th>mature_peptide_accession.version</th>\n",
       "      <th>mature_peptide_gi</th>\n",
       "      <th>Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>1246500</td>\n",
       "      <td>PROVISIONAL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NP_047184.1</td>\n",
       "      <td>10954455</td>\n",
       "      <td>NC_001911.1</td>\n",
       "      <td>10954454</td>\n",
       "      <td>348</td>\n",
       "      <td>1190</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>repA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>1246501</td>\n",
       "      <td>PROVISIONAL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NP_047186.1</td>\n",
       "      <td>10954457</td>\n",
       "      <td>NC_001911.1</td>\n",
       "      <td>10954454</td>\n",
       "      <td>2157</td>\n",
       "      <td>2912</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>repA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>1246502</td>\n",
       "      <td>VALIDATED</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NP_047187.1</td>\n",
       "      <td>10954458</td>\n",
       "      <td>NC_001911.1</td>\n",
       "      <td>10954454</td>\n",
       "      <td>3040</td>\n",
       "      <td>4590</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>leuA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>1246503</td>\n",
       "      <td>PROVISIONAL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NP_047188.1</td>\n",
       "      <td>10954459</td>\n",
       "      <td>NC_001911.1</td>\n",
       "      <td>10954454</td>\n",
       "      <td>4623</td>\n",
       "      <td>5714</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>leuB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>1246504</td>\n",
       "      <td>VALIDATED</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NP_047189.1</td>\n",
       "      <td>10954460</td>\n",
       "      <td>NC_001911.1</td>\n",
       "      <td>10954454</td>\n",
       "      <td>5717</td>\n",
       "      <td>7117</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>leuC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40141439</th>\n",
       "      <td>2749911</td>\n",
       "      <td>36959328</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NC_038002.1</td>\n",
       "      <td>1418587958</td>\n",
       "      <td>119755</td>\n",
       "      <td>119828</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>trnP-GGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40141440</th>\n",
       "      <td>2749911</td>\n",
       "      <td>36959329</td>\n",
       "      <td>PROVISIONAL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>YP_009496746.1</td>\n",
       "      <td>1418587959</td>\n",
       "      <td>NC_038002.1</td>\n",
       "      <td>1418587958</td>\n",
       "      <td>116</td>\n",
       "      <td>2374</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>psaA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40141441</th>\n",
       "      <td>2749911</td>\n",
       "      <td>36959330</td>\n",
       "      <td>PROVISIONAL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>YP_009496877.1</td>\n",
       "      <td>1418588090</td>\n",
       "      <td>NC_038002.1</td>\n",
       "      <td>1418587958</td>\n",
       "      <td>120098</td>\n",
       "      <td>120337</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>acpP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40141442</th>\n",
       "      <td>2749911</td>\n",
       "      <td>36959331</td>\n",
       "      <td>PROVISIONAL</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>YP_009496787.1</td>\n",
       "      <td>1418588000</td>\n",
       "      <td>NC_038002.1</td>\n",
       "      <td>1418587958</td>\n",
       "      <td>40018</td>\n",
       "      <td>40134</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>psbI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40141443</th>\n",
       "      <td>2749911</td>\n",
       "      <td>36959332</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NC_038002.1</td>\n",
       "      <td>1418587958</td>\n",
       "      <td>67821</td>\n",
       "      <td>69298</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rns</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>40141444 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          #tax_id    GeneID       status RNA_nucleotide_accession.version  \\\n",
       "0               9   1246500  PROVISIONAL                                -   \n",
       "1               9   1246501  PROVISIONAL                                -   \n",
       "2               9   1246502    VALIDATED                                -   \n",
       "3               9   1246503  PROVISIONAL                                -   \n",
       "4               9   1246504    VALIDATED                                -   \n",
       "...           ...       ...          ...                              ...   \n",
       "40141439  2749911  36959328          NaN                                -   \n",
       "40141440  2749911  36959329  PROVISIONAL                                -   \n",
       "40141441  2749911  36959330  PROVISIONAL                                -   \n",
       "40141442  2749911  36959331  PROVISIONAL                                -   \n",
       "40141443  2749911  36959332          NaN                                -   \n",
       "\n",
       "         RNA_nucleotide_gi protein_accession.version  protein_gi  \\\n",
       "0                        -               NP_047184.1    10954455   \n",
       "1                        -               NP_047186.1    10954457   \n",
       "2                        -               NP_047187.1    10954458   \n",
       "3                        -               NP_047188.1    10954459   \n",
       "4                        -               NP_047189.1    10954460   \n",
       "...                    ...                       ...         ...   \n",
       "40141439                 -                         -           -   \n",
       "40141440                 -            YP_009496746.1  1418587959   \n",
       "40141441                 -            YP_009496877.1  1418588090   \n",
       "40141442                 -            YP_009496787.1  1418588000   \n",
       "40141443                 -                         -           -   \n",
       "\n",
       "         genomic_nucleotide_accession.version genomic_nucleotide_gi  \\\n",
       "0                                 NC_001911.1              10954454   \n",
       "1                                 NC_001911.1              10954454   \n",
       "2                                 NC_001911.1              10954454   \n",
       "3                                 NC_001911.1              10954454   \n",
       "4                                 NC_001911.1              10954454   \n",
       "...                                       ...                   ...   \n",
       "40141439                          NC_038002.1            1418587958   \n",
       "40141440                          NC_038002.1            1418587958   \n",
       "40141441                          NC_038002.1            1418587958   \n",
       "40141442                          NC_038002.1            1418587958   \n",
       "40141443                          NC_038002.1            1418587958   \n",
       "\n",
       "         start_position_on_the_genomic_accession  \\\n",
       "0                                            348   \n",
       "1                                           2157   \n",
       "2                                           3040   \n",
       "3                                           4623   \n",
       "4                                           5717   \n",
       "...                                          ...   \n",
       "40141439                                  119755   \n",
       "40141440                                     116   \n",
       "40141441                                  120098   \n",
       "40141442                                   40018   \n",
       "40141443                                   67821   \n",
       "\n",
       "         end_position_on_the_genomic_accession orientation assembly  \\\n",
       "0                                         1190           -        -   \n",
       "1                                         2912           +        -   \n",
       "2                                         4590           +        -   \n",
       "3                                         5714           +        -   \n",
       "4                                         7117           +        -   \n",
       "...                                        ...         ...      ...   \n",
       "40141439                                119828           -        -   \n",
       "40141440                                  2374           +        -   \n",
       "40141441                                120337           -        -   \n",
       "40141442                                 40134           +        -   \n",
       "40141443                                 69298           +        -   \n",
       "\n",
       "         mature_peptide_accession.version mature_peptide_gi    Symbol  \n",
       "0                                       -                 -     repA1  \n",
       "1                                       -                 -     repA2  \n",
       "2                                       -                 -      leuA  \n",
       "3                                       -                 -      leuB  \n",
       "4                                       -                 -      leuC  \n",
       "...                                   ...               ...       ...  \n",
       "40141439                                -                 -  trnP-GGG  \n",
       "40141440                                -                 -      psaA  \n",
       "40141441                                -                 -      acpP  \n",
       "40141442                                -                 -      psbI  \n",
       "40141443                                -                 -       rns  \n",
       "\n",
       "[40141444 rows x 16 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "refseq = pd.read_table('gene2refseq')\n",
    "refseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#tax_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>PubMed_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>1246500</td>\n",
       "      <td>9873079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>1246501</td>\n",
       "      <td>9873079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>1246502</td>\n",
       "      <td>9873079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>1246503</td>\n",
       "      <td>9873079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>1246504</td>\n",
       "      <td>9873079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12757291</th>\n",
       "      <td>2747817</td>\n",
       "      <td>55898830</td>\n",
       "      <td>19273270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12757292</th>\n",
       "      <td>2747817</td>\n",
       "      <td>55898833</td>\n",
       "      <td>22037399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12757293</th>\n",
       "      <td>2747817</td>\n",
       "      <td>55898833</td>\n",
       "      <td>23651326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12757294</th>\n",
       "      <td>2747817</td>\n",
       "      <td>55898833</td>\n",
       "      <td>26712937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12757295</th>\n",
       "      <td>2747817</td>\n",
       "      <td>55898833</td>\n",
       "      <td>28069824</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12757296 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          #tax_id    GeneID  PubMed_ID\n",
       "0               9   1246500    9873079\n",
       "1               9   1246501    9873079\n",
       "2               9   1246502    9873079\n",
       "3               9   1246503    9873079\n",
       "4               9   1246504    9873079\n",
       "...           ...       ...        ...\n",
       "12757291  2747817  55898830   19273270\n",
       "12757292  2747817  55898833   22037399\n",
       "12757293  2747817  55898833   23651326\n",
       "12757294  2747817  55898833   26712937\n",
       "12757295  2747817  55898833   28069824\n",
       "\n",
       "[12757296 rows x 3 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pubmed = pd.read_table('gene2pubmed')\n",
    "pubmed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#tax_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Ensembl_gene_identifier</th>\n",
       "      <th>RNA_nucleotide_accession.version</th>\n",
       "      <th>Ensembl_rna_identifier</th>\n",
       "      <th>protein_accession.version</th>\n",
       "      <th>Ensembl_protein_identifier</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7227</td>\n",
       "      <td>30970</td>\n",
       "      <td>FBgn0040373</td>\n",
       "      <td>NM_001297803.1</td>\n",
       "      <td>FBtr0340207</td>\n",
       "      <td>NP_001284732.1</td>\n",
       "      <td>FBpp0309182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7227</td>\n",
       "      <td>30970</td>\n",
       "      <td>FBgn0040373</td>\n",
       "      <td>NM_130477.4</td>\n",
       "      <td>FBtr0070108</td>\n",
       "      <td>NP_569833.1</td>\n",
       "      <td>FBpp0070103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7227</td>\n",
       "      <td>30970</td>\n",
       "      <td>FBgn0040373</td>\n",
       "      <td>NM_166834.2</td>\n",
       "      <td>FBtr0070107</td>\n",
       "      <td>NP_726658.1</td>\n",
       "      <td>FBpp0070102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>7227</td>\n",
       "      <td>30971</td>\n",
       "      <td>FBgn0040372</td>\n",
       "      <td>NM_001272159.1</td>\n",
       "      <td>FBtr0332992</td>\n",
       "      <td>NP_001259088.1</td>\n",
       "      <td>FBpp0305207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7227</td>\n",
       "      <td>30971</td>\n",
       "      <td>FBgn0040372</td>\n",
       "      <td>NM_130478.3</td>\n",
       "      <td>FBtr0070063</td>\n",
       "      <td>NP_569834.1</td>\n",
       "      <td>FBpp0070062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3402972</th>\n",
       "      <td>2587831</td>\n",
       "      <td>115271344</td>\n",
       "      <td>ENSTMTG00000012187</td>\n",
       "      <td>XM_029913242.1</td>\n",
       "      <td>ENSTMTT00000017225.1</td>\n",
       "      <td>XP_029769102.1</td>\n",
       "      <td>ENSTMTP00000016631.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3402973</th>\n",
       "      <td>2587831</td>\n",
       "      <td>115271400</td>\n",
       "      <td>ENSTMTG00000007221</td>\n",
       "      <td>XM_029913853.1</td>\n",
       "      <td>ENSTMTT00000010245.1</td>\n",
       "      <td>XP_029769713.1</td>\n",
       "      <td>ENSTMTP00000009912.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3402974</th>\n",
       "      <td>2587831</td>\n",
       "      <td>115271535</td>\n",
       "      <td>ENSTMTG00000012454</td>\n",
       "      <td>XM_029914354.1</td>\n",
       "      <td>ENSTMTT00000017564.1</td>\n",
       "      <td>XP_029770214.1</td>\n",
       "      <td>ENSTMTP00000016963.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3402975</th>\n",
       "      <td>2587831</td>\n",
       "      <td>115271563</td>\n",
       "      <td>ENSTMTG00000012937</td>\n",
       "      <td>XM_029914401.1</td>\n",
       "      <td>ENSTMTT00000018218.1</td>\n",
       "      <td>XP_029770261.1</td>\n",
       "      <td>ENSTMTP00000017593.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3402976</th>\n",
       "      <td>2587831</td>\n",
       "      <td>115271572</td>\n",
       "      <td>ENSTMTG00000012792</td>\n",
       "      <td>XM_029914419.1</td>\n",
       "      <td>ENSTMTT00000018020.1</td>\n",
       "      <td>XP_029770279.1</td>\n",
       "      <td>ENSTMTP00000017406.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3402977 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #tax_id     GeneID Ensembl_gene_identifier  \\\n",
       "0           7227      30970             FBgn0040373   \n",
       "1           7227      30970             FBgn0040373   \n",
       "2           7227      30970             FBgn0040373   \n",
       "3           7227      30971             FBgn0040372   \n",
       "4           7227      30971             FBgn0040372   \n",
       "...          ...        ...                     ...   \n",
       "3402972  2587831  115271344      ENSTMTG00000012187   \n",
       "3402973  2587831  115271400      ENSTMTG00000007221   \n",
       "3402974  2587831  115271535      ENSTMTG00000012454   \n",
       "3402975  2587831  115271563      ENSTMTG00000012937   \n",
       "3402976  2587831  115271572      ENSTMTG00000012792   \n",
       "\n",
       "        RNA_nucleotide_accession.version Ensembl_rna_identifier  \\\n",
       "0                         NM_001297803.1            FBtr0340207   \n",
       "1                            NM_130477.4            FBtr0070108   \n",
       "2                            NM_166834.2            FBtr0070107   \n",
       "3                         NM_001272159.1            FBtr0332992   \n",
       "4                            NM_130478.3            FBtr0070063   \n",
       "...                                  ...                    ...   \n",
       "3402972                   XM_029913242.1   ENSTMTT00000017225.1   \n",
       "3402973                   XM_029913853.1   ENSTMTT00000010245.1   \n",
       "3402974                   XM_029914354.1   ENSTMTT00000017564.1   \n",
       "3402975                   XM_029914401.1   ENSTMTT00000018218.1   \n",
       "3402976                   XM_029914419.1   ENSTMTT00000018020.1   \n",
       "\n",
       "        protein_accession.version Ensembl_protein_identifier  \n",
       "0                  NP_001284732.1                FBpp0309182  \n",
       "1                     NP_569833.1                FBpp0070103  \n",
       "2                     NP_726658.1                FBpp0070102  \n",
       "3                  NP_001259088.1                FBpp0305207  \n",
       "4                     NP_569834.1                FBpp0070062  \n",
       "...                           ...                        ...  \n",
       "3402972            XP_029769102.1       ENSTMTP00000016631.1  \n",
       "3402973            XP_029769713.1       ENSTMTP00000009912.1  \n",
       "3402974            XP_029770214.1       ENSTMTP00000016963.1  \n",
       "3402975            XP_029770261.1       ENSTMTP00000017593.1  \n",
       "3402976            XP_029770279.1       ENSTMTP00000017406.1  \n",
       "\n",
       "[3402977 rows x 7 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensembl = pd.read_table('gene2ensembl')\n",
    "ensembl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#tax_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>LocusTag</th>\n",
       "      <th>Synonyms</th>\n",
       "      <th>dbXrefs</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>map_location</th>\n",
       "      <th>description</th>\n",
       "      <th>type_of_gene</th>\n",
       "      <th>Symbol_from_nomenclature_authority</th>\n",
       "      <th>Full_name_from_nomenclature_authority</th>\n",
       "      <th>Nomenclature_status</th>\n",
       "      <th>Other_designations</th>\n",
       "      <th>Modification_date</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Aliases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9606</td>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-</td>\n",
       "      <td>A1B|ABG|GAB|HYST2477</td>\n",
       "      <td>MIM:138670|HGNC:HGNC:5|Ensembl:ENSG00000121410</td>\n",
       "      <td>19</td>\n",
       "      <td>19q13.43</td>\n",
       "      <td>alpha-1-B glycoprotein</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>alpha-1-B glycoprotein</td>\n",
       "      <td>O</td>\n",
       "      <td>alpha-1B-glycoprotein|HEL-S-163pA|epididymis s...</td>\n",
       "      <td>20200313</td>\n",
       "      <td>-</td>\n",
       "      <td>[A1B, ABG, GAB, HYST2477, A1BG]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9606</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>-</td>\n",
       "      <td>A2MD|CPAMD5|FWP007|S863-7</td>\n",
       "      <td>MIM:103950|HGNC:HGNC:7|Ensembl:ENSG00000175899</td>\n",
       "      <td>12</td>\n",
       "      <td>12p13.31</td>\n",
       "      <td>alpha-2-macroglobulin</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>A2M</td>\n",
       "      <td>alpha-2-macroglobulin</td>\n",
       "      <td>O</td>\n",
       "      <td>alpha-2-macroglobulin|C3 and PZP-like alpha-2-...</td>\n",
       "      <td>20200510</td>\n",
       "      <td>-</td>\n",
       "      <td>[A2MD, CPAMD5, FWP007, S863-7, A2M]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9606</td>\n",
       "      <td>3</td>\n",
       "      <td>A2MP1</td>\n",
       "      <td>-</td>\n",
       "      <td>A2MP</td>\n",
       "      <td>HGNC:HGNC:8|Ensembl:ENSG00000256069</td>\n",
       "      <td>12</td>\n",
       "      <td>12p13.31</td>\n",
       "      <td>alpha-2-macroglobulin pseudogene 1</td>\n",
       "      <td>pseudo</td>\n",
       "      <td>A2MP1</td>\n",
       "      <td>alpha-2-macroglobulin pseudogene 1</td>\n",
       "      <td>O</td>\n",
       "      <td>pregnancy-zone protein pseudogene</td>\n",
       "      <td>20200313</td>\n",
       "      <td>-</td>\n",
       "      <td>[A2MP, A2MP1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9606</td>\n",
       "      <td>9</td>\n",
       "      <td>NAT1</td>\n",
       "      <td>-</td>\n",
       "      <td>AAC1|MNAT|NAT-1|NATI</td>\n",
       "      <td>MIM:108345|HGNC:HGNC:7645|Ensembl:ENSG00000171428</td>\n",
       "      <td>8</td>\n",
       "      <td>8p22</td>\n",
       "      <td>N-acetyltransferase 1</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NAT1</td>\n",
       "      <td>N-acetyltransferase 1</td>\n",
       "      <td>O</td>\n",
       "      <td>arylamine N-acetyltransferase 1|N-acetyltransf...</td>\n",
       "      <td>20200313</td>\n",
       "      <td>-</td>\n",
       "      <td>[AAC1, MNAT, NAT-1, NATI, NAT1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9606</td>\n",
       "      <td>10</td>\n",
       "      <td>NAT2</td>\n",
       "      <td>-</td>\n",
       "      <td>AAC2|NAT-2|PNAT</td>\n",
       "      <td>MIM:612182|HGNC:HGNC:7646|Ensembl:ENSG00000156006</td>\n",
       "      <td>8</td>\n",
       "      <td>8p22</td>\n",
       "      <td>N-acetyltransferase 2</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NAT2</td>\n",
       "      <td>N-acetyltransferase 2</td>\n",
       "      <td>O</td>\n",
       "      <td>arylamine N-acetyltransferase 2|N-acetyltransf...</td>\n",
       "      <td>20200405</td>\n",
       "      <td>-</td>\n",
       "      <td>[AAC2, NAT-2, PNAT, NAT2]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61625</th>\n",
       "      <td>741158</td>\n",
       "      <td>8923215</td>\n",
       "      <td>trnD</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>MT</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>20180406</td>\n",
       "      <td>-</td>\n",
       "      <td>[-, trnD]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61626</th>\n",
       "      <td>741158</td>\n",
       "      <td>8923216</td>\n",
       "      <td>trnP</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>MT</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>20180406</td>\n",
       "      <td>-</td>\n",
       "      <td>[-, trnP]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61627</th>\n",
       "      <td>741158</td>\n",
       "      <td>8923217</td>\n",
       "      <td>trnA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>MT</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>20180406</td>\n",
       "      <td>-</td>\n",
       "      <td>[-, trnA]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61628</th>\n",
       "      <td>741158</td>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>MT</td>\n",
       "      <td>-</td>\n",
       "      <td>cytochrome c oxidase subunit I</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>cytochrome c oxidase subunit I</td>\n",
       "      <td>20180406</td>\n",
       "      <td>-</td>\n",
       "      <td>[-, COX1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61629</th>\n",
       "      <td>741158</td>\n",
       "      <td>8923219</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>MT</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>20180406</td>\n",
       "      <td>-</td>\n",
       "      <td>[-, 16S rRNA]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61630 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #tax_id   GeneID    Symbol LocusTag                   Synonyms  \\\n",
       "0         9606        1      A1BG        -       A1B|ABG|GAB|HYST2477   \n",
       "1         9606        2       A2M        -  A2MD|CPAMD5|FWP007|S863-7   \n",
       "2         9606        3     A2MP1        -                       A2MP   \n",
       "3         9606        9      NAT1        -       AAC1|MNAT|NAT-1|NATI   \n",
       "4         9606       10      NAT2        -            AAC2|NAT-2|PNAT   \n",
       "...        ...      ...       ...      ...                        ...   \n",
       "61625   741158  8923215      trnD        -                          -   \n",
       "61626   741158  8923216      trnP        -                          -   \n",
       "61627   741158  8923217      trnA        -                          -   \n",
       "61628   741158  8923218      COX1        -                          -   \n",
       "61629   741158  8923219  16S rRNA        -                          -   \n",
       "\n",
       "                                                 dbXrefs chromosome  \\\n",
       "0         MIM:138670|HGNC:HGNC:5|Ensembl:ENSG00000121410         19   \n",
       "1         MIM:103950|HGNC:HGNC:7|Ensembl:ENSG00000175899         12   \n",
       "2                    HGNC:HGNC:8|Ensembl:ENSG00000256069         12   \n",
       "3      MIM:108345|HGNC:HGNC:7645|Ensembl:ENSG00000171428          8   \n",
       "4      MIM:612182|HGNC:HGNC:7646|Ensembl:ENSG00000156006          8   \n",
       "...                                                  ...        ...   \n",
       "61625                                                  -         MT   \n",
       "61626                                                  -         MT   \n",
       "61627                                                  -         MT   \n",
       "61628                                                  -         MT   \n",
       "61629                                                  -         MT   \n",
       "\n",
       "      map_location                         description    type_of_gene  \\\n",
       "0         19q13.43              alpha-1-B glycoprotein  protein-coding   \n",
       "1         12p13.31               alpha-2-macroglobulin  protein-coding   \n",
       "2         12p13.31  alpha-2-macroglobulin pseudogene 1          pseudo   \n",
       "3             8p22               N-acetyltransferase 1  protein-coding   \n",
       "4             8p22               N-acetyltransferase 2  protein-coding   \n",
       "...            ...                                 ...             ...   \n",
       "61625            -                                   -            tRNA   \n",
       "61626            -                                   -            tRNA   \n",
       "61627            -                                   -            tRNA   \n",
       "61628            -      cytochrome c oxidase subunit I  protein-coding   \n",
       "61629            -                                   -            rRNA   \n",
       "\n",
       "      Symbol_from_nomenclature_authority  \\\n",
       "0                                   A1BG   \n",
       "1                                    A2M   \n",
       "2                                  A2MP1   \n",
       "3                                   NAT1   \n",
       "4                                   NAT2   \n",
       "...                                  ...   \n",
       "61625                                  -   \n",
       "61626                                  -   \n",
       "61627                                  -   \n",
       "61628                                  -   \n",
       "61629                                  -   \n",
       "\n",
       "      Full_name_from_nomenclature_authority Nomenclature_status  \\\n",
       "0                    alpha-1-B glycoprotein                   O   \n",
       "1                     alpha-2-macroglobulin                   O   \n",
       "2        alpha-2-macroglobulin pseudogene 1                   O   \n",
       "3                     N-acetyltransferase 1                   O   \n",
       "4                     N-acetyltransferase 2                   O   \n",
       "...                                     ...                 ...   \n",
       "61625                                     -                   -   \n",
       "61626                                     -                   -   \n",
       "61627                                     -                   -   \n",
       "61628                                     -                   -   \n",
       "61629                                     -                   -   \n",
       "\n",
       "                                      Other_designations  Modification_date  \\\n",
       "0      alpha-1B-glycoprotein|HEL-S-163pA|epididymis s...           20200313   \n",
       "1      alpha-2-macroglobulin|C3 and PZP-like alpha-2-...           20200510   \n",
       "2                      pregnancy-zone protein pseudogene           20200313   \n",
       "3      arylamine N-acetyltransferase 1|N-acetyltransf...           20200313   \n",
       "4      arylamine N-acetyltransferase 2|N-acetyltransf...           20200405   \n",
       "...                                                  ...                ...   \n",
       "61625                                                  -           20180406   \n",
       "61626                                                  -           20180406   \n",
       "61627                                                  -           20180406   \n",
       "61628                     cytochrome c oxidase subunit I           20180406   \n",
       "61629                                                  -           20180406   \n",
       "\n",
       "      Feature_type                              Aliases  \n",
       "0                -      [A1B, ABG, GAB, HYST2477, A1BG]  \n",
       "1                -  [A2MD, CPAMD5, FWP007, S863-7, A2M]  \n",
       "2                -                        [A2MP, A2MP1]  \n",
       "3                -      [AAC1, MNAT, NAT-1, NATI, NAT1]  \n",
       "4                -            [AAC2, NAT-2, PNAT, NAT2]  \n",
       "...            ...                                  ...  \n",
       "61625            -                            [-, trnD]  \n",
       "61626            -                            [-, trnP]  \n",
       "61627            -                            [-, trnA]  \n",
       "61628            -                            [-, COX1]  \n",
       "61629            -                        [-, 16S rRNA]  \n",
       "\n",
       "[61630 rows x 17 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human = pd.read_table('Homo_sapiens.gene_info')\n",
    "\n",
    "\n",
    "aliases = human['Synonyms'].tolist()\n",
    "aliases_sep = []\n",
    "for i in range(len(aliases)):\n",
    "    my_list = aliases[i].split(\"|\")\n",
    "    my_list.append(human['Symbol'].iloc[i])\n",
    "    aliases_sep.append(my_list)\n",
    "human['Aliases'] = aliases_sep\n",
    "\n",
    "human"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>Gene stable ID version</th>\n",
       "      <th>Transcript stable ID</th>\n",
       "      <th>Transcript stable ID version</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>HGNC symbol</th>\n",
       "      <th>Gene start (bp)</th>\n",
       "      <th>Gene end (bp)</th>\n",
       "      <th>Transcript start (bp)</th>\n",
       "      <th>Transcript end (bp)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>ENSG00000210049.1</td>\n",
       "      <td>ENST00000387314</td>\n",
       "      <td>ENST00000387314.1</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>577</td>\n",
       "      <td>647</td>\n",
       "      <td>577</td>\n",
       "      <td>647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>ENSG00000211459.2</td>\n",
       "      <td>ENST00000389680</td>\n",
       "      <td>ENST00000389680.2</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>648</td>\n",
       "      <td>1601</td>\n",
       "      <td>648</td>\n",
       "      <td>1601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000210077</td>\n",
       "      <td>ENSG00000210077.1</td>\n",
       "      <td>ENST00000387342</td>\n",
       "      <td>ENST00000387342.1</td>\n",
       "      <td>MT-TV</td>\n",
       "      <td>MT-TV</td>\n",
       "      <td>1602</td>\n",
       "      <td>1670</td>\n",
       "      <td>1602</td>\n",
       "      <td>1670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000210082</td>\n",
       "      <td>ENSG00000210082.2</td>\n",
       "      <td>ENST00000387347</td>\n",
       "      <td>ENST00000387347.2</td>\n",
       "      <td>MT-RNR2</td>\n",
       "      <td>MT-RNR2</td>\n",
       "      <td>1671</td>\n",
       "      <td>3229</td>\n",
       "      <td>1671</td>\n",
       "      <td>3229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000209082</td>\n",
       "      <td>ENSG00000209082.1</td>\n",
       "      <td>ENST00000386347</td>\n",
       "      <td>ENST00000386347.1</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>3230</td>\n",
       "      <td>3304</td>\n",
       "      <td>3230</td>\n",
       "      <td>3304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249784</th>\n",
       "      <td>ENSG00000284869</td>\n",
       "      <td>ENSG00000284869.2</td>\n",
       "      <td>ENST00000644579</td>\n",
       "      <td>ENST00000644579.2</td>\n",
       "      <td>EEFSEC</td>\n",
       "      <td>EEFSEC</td>\n",
       "      <td>128160388</td>\n",
       "      <td>128415576</td>\n",
       "      <td>128160388</td>\n",
       "      <td>128415576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249785</th>\n",
       "      <td>ENSG00000284869</td>\n",
       "      <td>ENSG00000284869.2</td>\n",
       "      <td>ENST00000644955</td>\n",
       "      <td>ENST00000644955.1</td>\n",
       "      <td>EEFSEC</td>\n",
       "      <td>EEFSEC</td>\n",
       "      <td>128160388</td>\n",
       "      <td>128415576</td>\n",
       "      <td>128160408</td>\n",
       "      <td>128415572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249786</th>\n",
       "      <td>ENSG00000284869</td>\n",
       "      <td>ENSG00000284869.2</td>\n",
       "      <td>ENST00000645384</td>\n",
       "      <td>ENST00000645384.1</td>\n",
       "      <td>EEFSEC</td>\n",
       "      <td>EEFSEC</td>\n",
       "      <td>128160388</td>\n",
       "      <td>128415576</td>\n",
       "      <td>128160575</td>\n",
       "      <td>128348438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249787</th>\n",
       "      <td>ENSG00000284869</td>\n",
       "      <td>ENSG00000284869.2</td>\n",
       "      <td>ENST00000646013</td>\n",
       "      <td>ENST00000646013.1</td>\n",
       "      <td>EEFSEC</td>\n",
       "      <td>EEFSEC</td>\n",
       "      <td>128160388</td>\n",
       "      <td>128415576</td>\n",
       "      <td>128343834</td>\n",
       "      <td>128348718</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249788</th>\n",
       "      <td>ENSG00000285949</td>\n",
       "      <td>ENSG00000285949.1</td>\n",
       "      <td>ENST00000648703</td>\n",
       "      <td>ENST00000648703.1</td>\n",
       "      <td>AC084784.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>128188319</td>\n",
       "      <td>128201362</td>\n",
       "      <td>128188319</td>\n",
       "      <td>128201362</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>249789 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene stable ID Gene stable ID version Transcript stable ID  \\\n",
       "0       ENSG00000210049      ENSG00000210049.1      ENST00000387314   \n",
       "1       ENSG00000211459      ENSG00000211459.2      ENST00000389680   \n",
       "2       ENSG00000210077      ENSG00000210077.1      ENST00000387342   \n",
       "3       ENSG00000210082      ENSG00000210082.2      ENST00000387347   \n",
       "4       ENSG00000209082      ENSG00000209082.1      ENST00000386347   \n",
       "...                 ...                    ...                  ...   \n",
       "249784  ENSG00000284869      ENSG00000284869.2      ENST00000644579   \n",
       "249785  ENSG00000284869      ENSG00000284869.2      ENST00000644955   \n",
       "249786  ENSG00000284869      ENSG00000284869.2      ENST00000645384   \n",
       "249787  ENSG00000284869      ENSG00000284869.2      ENST00000646013   \n",
       "249788  ENSG00000285949      ENSG00000285949.1      ENST00000648703   \n",
       "\n",
       "       Transcript stable ID version   Gene name HGNC symbol  Gene start (bp)  \\\n",
       "0                 ENST00000387314.1       MT-TF       MT-TF              577   \n",
       "1                 ENST00000389680.2     MT-RNR1     MT-RNR1              648   \n",
       "2                 ENST00000387342.1       MT-TV       MT-TV             1602   \n",
       "3                 ENST00000387347.2     MT-RNR2     MT-RNR2             1671   \n",
       "4                 ENST00000386347.1      MT-TL1      MT-TL1             3230   \n",
       "...                             ...         ...         ...              ...   \n",
       "249784            ENST00000644579.2      EEFSEC      EEFSEC        128160388   \n",
       "249785            ENST00000644955.1      EEFSEC      EEFSEC        128160388   \n",
       "249786            ENST00000645384.1      EEFSEC      EEFSEC        128160388   \n",
       "249787            ENST00000646013.1      EEFSEC      EEFSEC        128160388   \n",
       "249788            ENST00000648703.1  AC084784.1         NaN        128188319   \n",
       "\n",
       "        Gene end (bp)  Transcript start (bp)  Transcript end (bp)  \n",
       "0                 647                    577                  647  \n",
       "1                1601                    648                 1601  \n",
       "2                1670                   1602                 1670  \n",
       "3                3229                   1671                 3229  \n",
       "4                3304                   3230                 3304  \n",
       "...               ...                    ...                  ...  \n",
       "249784      128415576              128160388            128415576  \n",
       "249785      128415576              128160408            128415572  \n",
       "249786      128415576              128160575            128348438  \n",
       "249787      128415576              128343834            128348718  \n",
       "249788      128201362              128188319            128201362  \n",
       "\n",
       "[249789 rows x 10 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mart = pd.read_csv('mart_export.txt',delimiter = ',')\n",
    "mart"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Brendan\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3063: DtypeWarning: Columns (7) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#tax_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>genomic_accession.version</th>\n",
       "      <th>genomic_gi</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>orientation</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>GeneIDs_on_left</th>\n",
       "      <th>distance_to_left</th>\n",
       "      <th>GeneIDs_on_right</th>\n",
       "      <th>distance_to_right</th>\n",
       "      <th>overlapping_GeneIDs</th>\n",
       "      <th>assembly</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9770</td>\n",
       "      <td>807607</td>\n",
       "      <td>NC_001321.1</td>\n",
       "      <td>5819095</td>\n",
       "      <td>8388</td>\n",
       "      <td>9068</td>\n",
       "      <td>+</td>\n",
       "      <td>MT</td>\n",
       "      <td>807612|807613</td>\n",
       "      <td>234</td>\n",
       "      <td>807610</td>\n",
       "      <td>854</td>\n",
       "      <td>807608|807611</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9770</td>\n",
       "      <td>807608</td>\n",
       "      <td>NC_001321.1</td>\n",
       "      <td>5819095</td>\n",
       "      <td>9068</td>\n",
       "      <td>9853</td>\n",
       "      <td>+</td>\n",
       "      <td>MT</td>\n",
       "      <td>807611|807612</td>\n",
       "      <td>650</td>\n",
       "      <td>807610|807616|807617</td>\n",
       "      <td>69</td>\n",
       "      <td>807607</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9770</td>\n",
       "      <td>807609</td>\n",
       "      <td>NC_001321.1</td>\n",
       "      <td>5819095</td>\n",
       "      <td>4354</td>\n",
       "      <td>5397</td>\n",
       "      <td>+</td>\n",
       "      <td>MT</td>\n",
       "      <td>807618</td>\n",
       "      <td>209</td>\n",
       "      <td>807612|807613</td>\n",
       "      <td>384</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9770</td>\n",
       "      <td>807610</td>\n",
       "      <td>NC_001321.1</td>\n",
       "      <td>5819095</td>\n",
       "      <td>9922</td>\n",
       "      <td>10267</td>\n",
       "      <td>+</td>\n",
       "      <td>MT</td>\n",
       "      <td>807607|807608|807611</td>\n",
       "      <td>69</td>\n",
       "      <td>807614|807615|807616|807617</td>\n",
       "      <td>71</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9770</td>\n",
       "      <td>807611</td>\n",
       "      <td>NC_001321.1</td>\n",
       "      <td>5819095</td>\n",
       "      <td>8227</td>\n",
       "      <td>8418</td>\n",
       "      <td>+</td>\n",
       "      <td>MT</td>\n",
       "      <td>807612|807613</td>\n",
       "      <td>73</td>\n",
       "      <td>807608|807610</td>\n",
       "      <td>650</td>\n",
       "      <td>807607</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27241512</th>\n",
       "      <td>1710540</td>\n",
       "      <td>55900172</td>\n",
       "      <td>NZ_CP058335.1</td>\n",
       "      <td>1867135660</td>\n",
       "      <td>1203619</td>\n",
       "      <td>1204218</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>55900170|55900171</td>\n",
       "      <td>43</td>\n",
       "      <td>55900173|55900174</td>\n",
       "      <td>120</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27241513</th>\n",
       "      <td>1710540</td>\n",
       "      <td>55900173</td>\n",
       "      <td>NZ_CP058335.1</td>\n",
       "      <td>1867135660</td>\n",
       "      <td>1204338</td>\n",
       "      <td>1205720</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>55900171|55900172</td>\n",
       "      <td>120</td>\n",
       "      <td>55900174|55900175</td>\n",
       "      <td>4</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27241514</th>\n",
       "      <td>1710540</td>\n",
       "      <td>55900174</td>\n",
       "      <td>NZ_CP058335.1</td>\n",
       "      <td>1867135660</td>\n",
       "      <td>1205724</td>\n",
       "      <td>1206146</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>55900172|55900173</td>\n",
       "      <td>4</td>\n",
       "      <td>55900175|55900176</td>\n",
       "      <td>100</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27241515</th>\n",
       "      <td>1710540</td>\n",
       "      <td>55900175</td>\n",
       "      <td>NZ_CP058335.1</td>\n",
       "      <td>1867135660</td>\n",
       "      <td>1206246</td>\n",
       "      <td>1207034</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>55900173|55900174</td>\n",
       "      <td>100</td>\n",
       "      <td>55900176|55900177</td>\n",
       "      <td>247</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27241516</th>\n",
       "      <td>1710540</td>\n",
       "      <td>55900176</td>\n",
       "      <td>NZ_CP058335.1</td>\n",
       "      <td>1867135660</td>\n",
       "      <td>1207281</td>\n",
       "      <td>1207514</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>55900174|55900175</td>\n",
       "      <td>247</td>\n",
       "      <td>55900177|55900178|55900179</td>\n",
       "      <td>7</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27241517 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          #tax_id    GeneID genomic_accession.version  genomic_gi  \\\n",
       "0            9770    807607               NC_001321.1     5819095   \n",
       "1            9770    807608               NC_001321.1     5819095   \n",
       "2            9770    807609               NC_001321.1     5819095   \n",
       "3            9770    807610               NC_001321.1     5819095   \n",
       "4            9770    807611               NC_001321.1     5819095   \n",
       "...           ...       ...                       ...         ...   \n",
       "27241512  1710540  55900172             NZ_CP058335.1  1867135660   \n",
       "27241513  1710540  55900173             NZ_CP058335.1  1867135660   \n",
       "27241514  1710540  55900174             NZ_CP058335.1  1867135660   \n",
       "27241515  1710540  55900175             NZ_CP058335.1  1867135660   \n",
       "27241516  1710540  55900176             NZ_CP058335.1  1867135660   \n",
       "\n",
       "          start_position  end_position orientation chromosome  \\\n",
       "0                   8388          9068           +         MT   \n",
       "1                   9068          9853           +         MT   \n",
       "2                   4354          5397           +         MT   \n",
       "3                   9922         10267           +         MT   \n",
       "4                   8227          8418           +         MT   \n",
       "...                  ...           ...         ...        ...   \n",
       "27241512         1203619       1204218           -          -   \n",
       "27241513         1204338       1205720           +          -   \n",
       "27241514         1205724       1206146           -          -   \n",
       "27241515         1206246       1207034           -          -   \n",
       "27241516         1207281       1207514           +          -   \n",
       "\n",
       "               GeneIDs_on_left distance_to_left             GeneIDs_on_right  \\\n",
       "0                807612|807613              234                       807610   \n",
       "1                807611|807612              650         807610|807616|807617   \n",
       "2                       807618              209                807612|807613   \n",
       "3         807607|807608|807611               69  807614|807615|807616|807617   \n",
       "4                807612|807613               73                807608|807610   \n",
       "...                        ...              ...                          ...   \n",
       "27241512     55900170|55900171               43            55900173|55900174   \n",
       "27241513     55900171|55900172              120            55900174|55900175   \n",
       "27241514     55900172|55900173                4            55900175|55900176   \n",
       "27241515     55900173|55900174              100            55900176|55900177   \n",
       "27241516     55900174|55900175              247   55900177|55900178|55900179   \n",
       "\n",
       "         distance_to_right overlapping_GeneIDs assembly  \n",
       "0                      854       807608|807611        -  \n",
       "1                       69              807607        -  \n",
       "2                      384                   -        -  \n",
       "3                       71                   -        -  \n",
       "4                      650              807607        -  \n",
       "...                    ...                 ...      ...  \n",
       "27241512               120                   -        -  \n",
       "27241513                 4                   -        -  \n",
       "27241514               100                   -        -  \n",
       "27241515               247                   -        -  \n",
       "27241516                 7                   -        -  \n",
       "\n",
       "[27241517 rows x 14 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neighbors = pd.read_table('gene_neighbors')\n",
    "neighbors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#tax_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>relationship</th>\n",
       "      <th>Other_tax_id</th>\n",
       "      <th>Other_GeneID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7955</td>\n",
       "      <td>30037</td>\n",
       "      <td>Ortholog</td>\n",
       "      <td>7897</td>\n",
       "      <td>102366554</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7955</td>\n",
       "      <td>30037</td>\n",
       "      <td>Ortholog</td>\n",
       "      <td>7994</td>\n",
       "      <td>103038481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7955</td>\n",
       "      <td>30037</td>\n",
       "      <td>Ortholog</td>\n",
       "      <td>7998</td>\n",
       "      <td>108255935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>7955</td>\n",
       "      <td>30037</td>\n",
       "      <td>Ortholog</td>\n",
       "      <td>8005</td>\n",
       "      <td>113577924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7955</td>\n",
       "      <td>30037</td>\n",
       "      <td>Ortholog</td>\n",
       "      <td>8010</td>\n",
       "      <td>105014395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4771230</th>\n",
       "      <td>10090</td>\n",
       "      <td>113939900</td>\n",
       "      <td>Ortholog</td>\n",
       "      <td>9913</td>\n",
       "      <td>113939901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4771231</th>\n",
       "      <td>10090</td>\n",
       "      <td>113939900</td>\n",
       "      <td>Ortholog</td>\n",
       "      <td>10116</td>\n",
       "      <td>113939903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4771232</th>\n",
       "      <td>10090</td>\n",
       "      <td>113939900</td>\n",
       "      <td>Ortholog</td>\n",
       "      <td>27679</td>\n",
       "      <td>113939905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4771233</th>\n",
       "      <td>10090</td>\n",
       "      <td>113939900</td>\n",
       "      <td>Ortholog</td>\n",
       "      <td>28377</td>\n",
       "      <td>113939909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4771234</th>\n",
       "      <td>10090</td>\n",
       "      <td>113939900</td>\n",
       "      <td>Ortholog</td>\n",
       "      <td>1868482</td>\n",
       "      <td>113939906</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4771235 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #tax_id     GeneID relationship  Other_tax_id  Other_GeneID\n",
       "0           7955      30037     Ortholog          7897     102366554\n",
       "1           7955      30037     Ortholog          7994     103038481\n",
       "2           7955      30037     Ortholog          7998     108255935\n",
       "3           7955      30037     Ortholog          8005     113577924\n",
       "4           7955      30037     Ortholog          8010     105014395\n",
       "...          ...        ...          ...           ...           ...\n",
       "4771230    10090  113939900     Ortholog          9913     113939901\n",
       "4771231    10090  113939900     Ortholog         10116     113939903\n",
       "4771232    10090  113939900     Ortholog         27679     113939905\n",
       "4771233    10090  113939900     Ortholog         28377     113939909\n",
       "4771234    10090  113939900     Ortholog       1868482     113939906\n",
       "\n",
       "[4771235 rows x 5 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orthologs = pd.read_table('gene_orthologs')\n",
    "orthologs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>--------------------------------------------------------</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JrId: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>JournalTitle: AADE editors' journal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MedAbbr: AADE Ed J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ISSN (Print): 0160-6999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ISSN (Online):</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294986</th>\n",
       "      <td>MedAbbr: Acta Anal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294987</th>\n",
       "      <td>ISSN (Print): 0353-5150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294988</th>\n",
       "      <td>ISSN (Online): 1874-6349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294989</th>\n",
       "      <td>IsoAbbr: Acta Anal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294990</th>\n",
       "      <td>NlmId: 101769522</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>294991 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       --------------------------------------------------------\n",
       "0                                                 JrId: 1      \n",
       "1                     JournalTitle: AADE editors' journal      \n",
       "2                                      MedAbbr: AADE Ed J      \n",
       "3                                 ISSN (Print): 0160-6999      \n",
       "4                                         ISSN (Online):       \n",
       "...                                                   ...      \n",
       "294986                                 MedAbbr: Acta Anal      \n",
       "294987                            ISSN (Print): 0353-5150      \n",
       "294988                           ISSN (Online): 1874-6349      \n",
       "294989                                 IsoAbbr: Acta Anal      \n",
       "294990                                   NlmId: 101769522      \n",
       "\n",
       "[294991 rows x 1 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "j_entrez = pd.read_table('J_Entrez')\n",
    "j_entrez"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['#tax_id', 'GeneID', 'status', 'RNA_nucleotide_accession.version',\n",
      "       'RNA_nucleotide_gi', 'protein_accession.version', 'protein_gi',\n",
      "       'genomic_nucleotide_accession.version', 'genomic_nucleotide_gi',\n",
      "       'start_position_on_the_genomic_accession',\n",
      "       'end_position_on_the_genomic_accession', 'orientation', 'assembly',\n",
      "       'mature_peptide_accession.version', 'mature_peptide_gi', 'Symbol'],\n",
      "      dtype='object')\n",
      "Index(['#tax_id', 'GeneID', 'PubMed_ID'], dtype='object')\n",
      "Index(['#tax_id', 'GeneID', 'Ensembl_gene_identifier',\n",
      "       'RNA_nucleotide_accession.version', 'Ensembl_rna_identifier',\n",
      "       'protein_accession.version', 'Ensembl_protein_identifier'],\n",
      "      dtype='object')\n",
      "Index(['#tax_id', 'GeneID', 'Symbol', 'LocusTag', 'Synonyms', 'dbXrefs',\n",
      "       'chromosome', 'map_location', 'description', 'type_of_gene',\n",
      "       'Symbol_from_nomenclature_authority',\n",
      "       'Full_name_from_nomenclature_authority', 'Nomenclature_status',\n",
      "       'Other_designations', 'Modification_date', 'Feature_type'],\n",
      "      dtype='object')\n",
      "Index(['Gene stable ID', 'Gene stable ID version', 'Transcript stable ID',\n",
      "       'Transcript stable ID version', 'Gene name', 'HGNC symbol',\n",
      "       'Gene start (bp)', 'Gene end (bp)', 'Transcript start (bp)',\n",
      "       'Transcript end (bp)'],\n",
      "      dtype='object')\n",
      "Index(['#tax_id', 'GeneID', 'genomic_accession.version', 'genomic_gi',\n",
      "       'start_position', 'end_position', 'orientation', 'chromosome',\n",
      "       'GeneIDs_on_left', 'distance_to_left', 'GeneIDs_on_right',\n",
      "       'distance_to_right', 'overlapping_GeneIDs', 'assembly'],\n",
      "      dtype='object')\n",
      "Index(['#tax_id', 'GeneID', 'relationship', 'Other_tax_id', 'Other_GeneID'], dtype='object')\n",
      "Index(['--------------------------------------------------------'], dtype='object')\n"
     ]
    }
   ],
   "source": [
    "def compare_sets(list_data):\n",
    "    for ele in list_data:\n",
    "        col_names = ele.columns\n",
    "        print(col_names)\n",
    "data_list = refseq,pubmed,ensembl,human,mart,neighbors,orthologs,j_entrez\n",
    "compare_sets(data_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "def match_lists(data_list):\n",
    "    for i in range(len(data_list)):\n",
    "        for j in range(len(data_list)):\n",
    "            if (i != j and i < j):\n",
    "                ### translation 1 test col name\n",
    "                test_col1 = data_list[i][1]\n",
    "\n",
    "                ### translation 1 reference dataset (human,gene,mart)\n",
    "                ref_df1 = pd.read_table(data_list[j][0])\n",
    "\n",
    "                ### translation 1 reference col name\n",
    "                ref_col1 = data_list[j][1]\n",
    "\n",
    "                ### translation 1 output col name\n",
    "                out_col1 = 'MATCH'\n",
    "                \n",
    "                trans_list = [[test_col1,ref_df1,ref_col1,out_col1]]\n",
    "\n",
    "                match_percent = combined_translate(data_list[i][0],'other', trans_list)\n",
    "                print(match_percent,\" percent mismatch, \", data_list[i][0], 'to', data_list[j][0])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "96.6667  percent mismatch,  t2d_SNP.txt to t2dm.txt\n",
      "100.0  percent mismatch,  t2d_SNP.txt to diabetesassay1\n",
      "80.0  percent mismatch,  t2d_SNP.txt to association_study.txt\n",
      "100.0  percent mismatch,  t2d_SNP.txt to novel_t2d.txt\n",
      "100.0  percent mismatch,  t2dm.txt to diabetesassay1\n",
      "83.3333  percent mismatch,  t2dm.txt to association_study.txt\n",
      "100.0  percent mismatch,  t2dm.txt to novel_t2d.txt\n",
      "35.7143  percent mismatch,  diabetesassay1 to association_study.txt\n",
      "100.0  percent mismatch,  diabetesassay1 to novel_t2d.txt\n",
      "100.0  percent mismatch,  association_study.txt to novel_t2d.txt\n"
     ]
    }
   ],
   "source": [
    "t2d = pd.read_table('t2d_SNP.txt')\n",
    "\n",
    "# t2dm - https://www.genome.jp/kegg-bin/show_pathway?hsa04930\n",
    "t2d\n",
    "\n",
    "t2dm = pd.read_table('t2dm.txt')\n",
    "t2dm\n",
    "# https://www.genome.jp/kegg-bin/show_pathway?hsa04930\n",
    "\n",
    "da1 = pd.read_table('diabetesassay1')\n",
    "# https://www.nature.com/articles/s41467-018-06249-3\n",
    "da1\n",
    "\n",
    "ass_st = pd.read_table('association_study.txt')\n",
    "t2dm\n",
    "# https://onlinelibrary.wiley.com/doi/full/10.1111/j.1365-2362.2010.02454.x\n",
    "\n",
    "novel_t2d = pd.read_table('novel_t2d.txt')\n",
    "#https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003046\n",
    "\n",
    "l1 = ['t2d_SNP.txt', 'Gene symbol']\n",
    "l2 = ['t2dm.txt', 'Gene_name']\n",
    "l3 = ['diabetesassay1','Human ortholog']\n",
    "l4 = ['association_study.txt','Gene loci']\n",
    "l5 = ['novel_t2d.txt','Internode gene']\n",
    "match_lists([l1,l2,l3,l4,l5])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "def combined_translate(input_file, file_type, trans_list):\n",
    "    if (file_type == 'csv'):\n",
    "        df = pd.read_csv(input_file)\n",
    "    if (file_type == 'excel'):\n",
    "        df = pd.read_excel(input_file)\n",
    "    if (file_type == 'other'):\n",
    "        df = pd.read_table(input_file)\n",
    "    \n",
    "    for item in trans_list:\n",
    "        test_col = item[0]\n",
    "        reference_df = item[1]\n",
    "        reference_col = item[2]\n",
    "        output_col = item[3]\n",
    "        \n",
    "        \n",
    "        start_param = df[test_col].tolist()\n",
    "        in_list = reference_df[reference_col].tolist()\n",
    "        out_listoflist = []\n",
    "\n",
    "        for i in range(len(start_param)):\n",
    "            param = start_param[i]\n",
    "\n",
    "            match_list=[]\n",
    "            for j in range(len(in_list)):\n",
    "                if str(param) in str(in_list[j]):\n",
    "                    index_match = reference_df.iloc[j]\n",
    "                    match_list.append(index_match)\n",
    "\n",
    "            out_list = []\n",
    "            for k in range(len(match_list)):\n",
    "                if (output_col == 'MATCH'):\n",
    "                    out_list.append(match_list[k][reference_col])\n",
    "                else: \n",
    "                    out_list.append(match_list[k][output_col])\n",
    "\n",
    "            out_listoflist.append(out_list)\n",
    "        out_colname = 'output ' + reference_col\n",
    "        df[out_colname] = out_listoflist\n",
    "\n",
    "    \n",
    "        nonmatches_ind = [ind for ind, x in enumerate(df[out_colname]) if len(x)==0 or x != x or x=='-']\n",
    "        matches_ind = [ind for ind, x in enumerate(df[out_colname]) if len(x)!=0 or x == x or x!='-']\n",
    "        nonmatches = df.iloc[nonmatches_ind]\n",
    "        matches = df.iloc[matches_ind]\n",
    "        \n",
    "        return round(100*len(nonmatches)/len(df),4)\n",
    "#         print(round(100*len(nonmatches)/len(df),4),\" percent mismatch, \",input_file, 'to')\n",
    "    \n",
    "#     print(df)\n",
    "    \n",
    "        \n",
    "          "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5.6604  percent mismatch,  Gene loci\n",
      "        Gene loci                                     output Aliases\n",
      "0           PPARG  [[CIMT1, GLM1, NR1C3, PPARG1, PPARG2, PPARG5, ...\n",
      "1          KCNJ11  [[BIR, HHF2, IKATP, KIR6.2, MODY13, PHHI, PNDM...\n",
      "2          TCF7L2                            [[TCF-4, TCF4, TCF7L2]]\n",
      "3            WFS1                 [[CTRCT41, WFRS, WFS, WFSL, WFS1]]\n",
      "4           HNF1B  [[FJHN, HNF-1-beta, HNF-1B, HNF1beta, HNF2, HP...\n",
      "5             FTO  [[ALKBH9, BMIQ14, GDFD, FTO], [-, FTO-IT1], [-...\n",
      "6         SLC30A8                           [[ZNT8, ZnT-8, SLC30A8]]\n",
      "7            HHEX         [[HEX, HMPH, HOX11L-PEN, PRH, PRHX, HHEX]]\n",
      "8          CDKAL1                                      [[-, CDKAL1]]\n",
      "9         IGF2BP2  [[IMP-2, IMP2, VICKZ2, IGF2BP2], [C3orf65, IGF...\n",
      "10         CDKN2A  [[ARF, CDK4I, CDKN2, CMM2, INK4, INK4A, MLM, M...\n",
      "11         TSPAN8                         [[CO-029, TM4SF3, TSPAN8]]\n",
      "12        ADAMTS9  [[-, ADAMTS9], [-, ADAMTS9-AS2], [-, ADAMTS9-A...\n",
      "13         NOTCH2  [[AGS2, HJCYS, hN2, NOTCH2], [N2N, NOTCH2NL, N...\n",
      "14  CDC123-CAMK1D                                                 []\n",
      "15          THADA                            [[ARMC13, GITA, THADA]]\n",
      "16          JAZF1           [[TIP27, ZNF802, JAZF1], [-, JAZF1-AS1]]\n",
      "17          KCNQ1  [[ATFB1, ATFB3, JLNS1, KCNA8, KCNA9, KVLQT1, K...\n",
      "18           IRS1  [[HIRS-1, IRS1], [DIRS1, FNDC6, IL-20R2, IL20R...\n",
      "19          DUSP9                             [[MKP-4, MKP4, DUSP9]]\n",
      "20         ZFAND6   [[AWP1, ZA20D3, ZFAND5B, ZFAND6], [-, ZFAND6P1]]\n",
      "21           PRC1  [[ADPRC 1, ADPRC1, CD38], [GPRC1A, MGLU1, MGLU...\n",
      "22         CENTD2                                  [[CENTD2, ARAP1]]\n",
      "23       TP53INP1  [[SIP, TP53DINP1, TP53INP1A, TP53INP1B, Teap, ...\n",
      "24          KLF14    [[EPFN, EPIPROFIN, KLF14, SP6], [BTEB5, KLF14]]\n",
      "25          ZBED3                       [[-, ZBED3], [-, ZBED3-AS1]]\n",
      "26         BCL11A  [[BCL11A-L, BCL11A-S, BCL11A-XL, BCL11a-M, CTI...\n",
      "27          HNF1A  [[HNF-1A, HNF1, HNF4A, IDDM20, LFB1, MODY3, TC...\n",
      "28         CHCHD9                      [[C9orf49, CHCHD9, CHCHD2P9]]\n",
      "29          HMGA2  [[BABL, HMGI-C, HMGIC, LIPO, STQTL9, HMGA2], [...\n",
      "30         MTNR1B                  [[FGQTL2, MEL-1B-R, MT2, MTNR1B]]\n",
      "31   DGKB-TMEM195                                                 []\n",
      "32           GCKR  [[FGQTL5, GKRP, GCKR], [GCKR, KHS, KHS1, MAPKK...\n",
      "33      GCKRPROX1                                                 []\n",
      "34          ADCY5                               [[AC5, FDFM, ADCY5]]\n",
      "35         SLC2A2                                  [[GLUT2, SLC2A2]]\n",
      "36          G6PC2                                    [[IGRP, G6PC2]]\n",
      "37          GLIS3       [[C9orf70, GLIS3-AS1], [NDH, ZNF515, GLIS3]]\n",
      "38         ADRA2A  [[ADRA2, ADRA2R, ADRAR, ALPHA2AAR, ZNF32, ADRA...\n",
      "39           CRY2                             [[HCRY2, PHLL2, CRY2]]\n",
      "40           MADD  [[EMA, GA2, MADD, ETFA], [FP585, MADD, ETFB], ...\n",
      "41          FADS1  [[D5D, FADS6, FADSD5, LLCDL1, TU12, FADS1], [C...\n",
      "42           IGF1  [[IGF, IGF-I, IGFI, MGF, IGF1], [CD221, IGFIR,...\n",
      "43           GIPR                                   [[PGQTL2, GIPR]]\n",
      "44         VPS13C                                 [[PARK23, VPS13C]]\n",
      "45         C2CD4B                          [[FAM148B, NLF2, C2CD4B]]\n",
      "46            HK1  [[CHK1, CHEK1], [CBP, KAT3A, MKHK1, RSTS, RSTS...\n",
      "47           FN3K                       [[-, FN3K], [FN3KL, FN3KRP]]\n",
      "48            HFE  [[FHC, FTH, FTHL6, HFE5, PIG15, PLIF, FTH1], [...\n",
      "49        TMPRSS6                            [[IRIDA, MT2, TMPRSS6]]\n",
      "50           ANK1  [[ANK, SPH1, SPH2, ANK1], [ANKRD59, BAT4, D6S5...\n",
      "51          SPTA1               [[EL2, HPP, HS3, SPH3, SPTA, SPTA1]]\n",
      "52         ATP11A  [[ATPIH, ATPIS, ATP11A], [C13orf35, SMABLO1, A...\n"
     ]
    }
   ],
   "source": [
    "### Name of input data file\n",
    "input_file = 'association_study.txt'\n",
    "\n",
    "### Type of file (excel, csv, or other)\n",
    "file_type = 'other'\n",
    "\n",
    "### translation 1 test col name\n",
    "test_col1 = 'Gene loci'\n",
    "\n",
    "### translation 1 reference dataset (human,gene,mart)\n",
    "ref_df1 = human\n",
    "\n",
    "### translation 1 reference col name\n",
    "ref_col1 = 'Aliases'\n",
    "    \n",
    "### translation 1 output col name\n",
    "out_col1 = 'MATCH'\n",
    "\n",
    "### translation 1 use only if one translation, otherwise delete\n",
    "trans_list = [[test_col1,ref_df1,ref_col1,out_col1]]\n",
    "combined_translate(input_file,file_type,trans_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0  percent mismatch,  Gene symbol\n",
      "            SNP Gene symbol  Chromosome Position (bp)*  \\\n",
      "0     rs4740283     RAPGEF1           9    131,477,850   \n",
      "1    rs2021966†       ENPP1           6    132,192,132   \n",
      "2   rs1042522†‡        TP53          17      7,520,197   \n",
      "3     rs1882095        NRF1           7    128,991,595   \n",
      "4    rs10513684      SLC2A2           3    172,206,912   \n",
      "5     rs1801282       PPARG           3     12,368,125   \n",
      "6      rs222852      SLC2A4          17      7,081,330   \n",
      "7     rs4843165       FOXC2          16     85,162,542   \n",
      "8       rs5400‡      SLC2A2           3    172,215,002   \n",
      "9      rs858341       ENPP1           6    132,202,148   \n",
      "10    rs1349498     RAPGEF4           2    173,418,113   \n",
      "11    rs8069976       SOCS3          17     73,861,445   \n",
      "12    rs3769249     RAPGEF4           2    173,648,169   \n",
      "13   rs17280845        CBLB           3    106,927,226   \n",
      "14      rs5219‡      KCNJ11          11     17,366,148   \n",
      "15   rs10915239        MECR           1     29,344,565   \n",
      "16   rs11206883      PRKAA2           1     56,815,240   \n",
      "17   rs11183212       ARID2          12     44,500,134   \n",
      "18    rs2395836     PRKAR2B           7    106,381,475   \n",
      "19    rs2970871    PPARGC1A           4     23,566,851   \n",
      "20      rs11888        JAK3          19     17,796,626   \n",
      "21    rs2235718       FOXC1           6      1,552,602   \n",
      "22    rs3118526        RXRA           9    134,563,302   \n",
      "23       rs9313      SORBS1          10     97,061,862   \n",
      "24    rs9402346       ENPP1           6    132,185,981   \n",
      "25    rs1830971       ENPP1           6    132,190,046   \n",
      "26    rs1409184       ENPP1           6    132,182,184   \n",
      "27    rs6802898       PPARG           3     12,366,207   \n",
      "28    rs7796553        NRF1           7    128,974,194   \n",
      "29     rs943852     RAPGEF1           9                  \n",
      "\n",
      "                                       output Aliases  \n",
      "0                                              [2889]  \n",
      "1                                              [5167]  \n",
      "2   [2593, 5625, 6288, 7157, 7158, 7159, 8379, 850...  \n",
      "3                          [4779, 4899, 84937, 91694]  \n",
      "4                                              [6514]  \n",
      "5                         [5468, 5469, 10891, 133522]  \n",
      "6                                       [6517, 56731]  \n",
      "7                                   [2303, 103752587]  \n",
      "8                                              [6514]  \n",
      "9                                              [5167]  \n",
      "10                                     [11069, 91149]  \n",
      "11                                             [9021]  \n",
      "12                                     [11069, 91149]  \n",
      "13                                              [868]  \n",
      "14                                             [3767]  \n",
      "15  [9949, 51102, 83607, 128989, 284439, 100420921...  \n",
      "16                                             [5563]  \n",
      "17                          [3720, 196528, 100506681]  \n",
      "18                                  [5577, 105375441]  \n",
      "19                                            [10891]  \n",
      "20                                             [3718]  \n",
      "21                                             [2296]  \n",
      "22                                             [6256]  \n",
      "23                                            [10580]  \n",
      "24                                             [5167]  \n",
      "25                                             [5167]  \n",
      "26                                             [5167]  \n",
      "27                        [5468, 5469, 10891, 133522]  \n",
      "28                         [4779, 4899, 84937, 91694]  \n",
      "29                                             [2889]  \n"
     ]
    }
   ],
   "source": [
    "### Name of input data file\n",
    "input_file = 't2d_SNP.txt'\n",
    "\n",
    "### Type of file (excel, csv, or other)\n",
    "file_type = 'other'\n",
    "\n",
    "### translation 1 test col name\n",
    "test_col1 = 'Gene symbol'\n",
    "\n",
    "### translation 1 reference dataset (human,gene,mart)\n",
    "ref_df1 = human\n",
    "\n",
    "### translation 1 reference col name\n",
    "ref_col1 = 'Aliases'\n",
    "    \n",
    "### translation 1 output col name\n",
    "out_col1 = 'GeneID'\n",
    "\n",
    "### translation 1 use only if one translation, otherwise delete\n",
    "trans_list = [[test_col1,ref_df1,ref_col1,out_col1]]\n",
    "combined_translate(input_file,file_type,trans_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12.5  percent mismatch,  Gene_name\n",
      "   Gene_name                                     output Aliases\n",
      "0      ADIPO    [9370, 51094, 79602, 390503, 645220, 100874095]\n",
      "1      GLUT4                                             [6517]\n",
      "2        INS  [2305, 3342, 3416, 3630, 3638, 3640, 3641, 364...\n",
      "3       INSR                                       [3643, 3645]\n",
      "4       IRS1                      [3667, 53833, 112802, 147183]\n",
      "5        IRS  [2618, 3376, 3667, 8471, 8660, 10458, 53833, 5...\n",
      "6       P13K                                                 []\n",
      "7       SOCS  [1154, 8651, 8835, 9021, 9306, 9655, 30837, 12...\n",
      "8        ERK  [1942, 1943, 1944, 1945, 1946, 1947, 1948, 194...\n",
      "9        IKK           [1147, 3551, 8517, 9641, 117166, 124857]\n",
      "10       JNK              [5599, 5601, 5602, 5609, 6416, 23005]\n",
      "11       TNF  [355, 356, 608, 690, 939, 943, 944, 958, 959, ...\n",
      "12      TNFA  [1942, 5806, 7124, 7126, 7127, 7128, 7130, 713...\n",
      "13      mTOR                                                 []\n",
      "14     PRKCZ                                  [5590, 100506504]\n",
      "15     PRKCD                                     [5580, 112464]\n",
      "16      PDX1                            [3651, 8050, 100861550]\n",
      "17      MafA                                           [389692]\n",
      "18      VDCC                                                 []\n",
      "19      SUR1                                             [6833]\n",
      "20    KCNJ11                                             [3767]\n",
      "21     GLUT2                               [6514, 57084, 79070]\n",
      "22        GK  [780, 1606, 1607, 1608, 1609, 1716, 2584, 2585...\n",
      "23       PYK                                       [2185, 5256]\n"
     ]
    }
   ],
   "source": [
    "### Name of input data file\n",
    "input_file = 'T2dm.txt'\n",
    "\n",
    "### Type of file (excel, csv, or other)\n",
    "file_type = 'other'\n",
    "\n",
    "### translation 1 test col name\n",
    "test_col1 = 'Gene_name'\n",
    "\n",
    "### translation 1 reference dataset (human,gene,mart)\n",
    "ref_df1 = human\n",
    "\n",
    "### translation 1 reference col name\n",
    "ref_col1 = 'Aliases'\n",
    "    \n",
    "### translation 1 output col name\n",
    "out_col1 = 'GeneID'\n",
    "\n",
    "### translation 1 use only if one translation, otherwise delete\n",
    "trans_list = [[test_col1,ref_df1,ref_col1,out_col1]]\n",
    "combined_translate(input_file,file_type,trans_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "l1 = []"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
