{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Errno 2] No such file or directory: 'C:UsersBrendanDocumentsSummer20'\n",
      "/Users/bkearney/Summer_20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/bkearney/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:20: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n",
      "/Users/bkearney/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:21: FutureWarning: read_table is deprecated, use read_csv instead, passing sep='\\t'.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#tax_id</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>LocusTag</th>\n",
       "      <th>Synonyms</th>\n",
       "      <th>dbXrefs</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>map_location</th>\n",
       "      <th>description</th>\n",
       "      <th>type_of_gene</th>\n",
       "      <th>Symbol_from_nomenclature_authority</th>\n",
       "      <th>Full_name_from_nomenclature_authority</th>\n",
       "      <th>Nomenclature_status</th>\n",
       "      <th>Other_designations</th>\n",
       "      <th>Modification_date</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Aliases</th>\n",
       "      <th>geneId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22260</th>\n",
       "      <td>9606</td>\n",
       "      <td>353500</td>\n",
       "      <td>BMP8A</td>\n",
       "      <td>-</td>\n",
       "      <td>OP-2</td>\n",
       "      <td>HGNC:HGNC:21650|Ensembl:ENSG00000183682</td>\n",
       "      <td>1</td>\n",
       "      <td>1p34.3</td>\n",
       "      <td>bone morphogenetic protein 8a</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>BMP8A</td>\n",
       "      <td>bone morphogenetic protein 8a</td>\n",
       "      <td>O</td>\n",
       "      <td>bone morphogenetic protein 8A|BMP-8A</td>\n",
       "      <td>20200313</td>\n",
       "      <td>-</td>\n",
       "      <td>[OP-2, BMP8A]</td>\n",
       "      <td>ENSG00000183682</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       #tax_id  GeneID Symbol LocusTag Synonyms  \\\n",
       "22260     9606  353500  BMP8A        -     OP-2   \n",
       "\n",
       "                                       dbXrefs chromosome map_location  \\\n",
       "22260  HGNC:HGNC:21650|Ensembl:ENSG00000183682          1       1p34.3   \n",
       "\n",
       "                         description    type_of_gene  \\\n",
       "22260  bone morphogenetic protein 8a  protein-coding   \n",
       "\n",
       "      Symbol_from_nomenclature_authority  \\\n",
       "22260                              BMP8A   \n",
       "\n",
       "      Full_name_from_nomenclature_authority Nomenclature_status  \\\n",
       "22260         bone morphogenetic protein 8a                   O   \n",
       "\n",
       "                         Other_designations  Modification_date Feature_type  \\\n",
       "22260  bone morphogenetic protein 8A|BMP-8A           20200313            -   \n",
       "\n",
       "             Aliases           geneId  \n",
       "22260  [OP-2, BMP8A]  ENSG00000183682  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"\n",
    "\n",
    "@author: bkearney\n",
    "email: bkearne5@uncc.edu\n",
    "\n",
    "Purpose: improve from TranslationfromCSV code with new datasets and more ubiquitous code\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import re\n",
    "import os\n",
    "import statistics \n",
    "# import allel\n",
    "\n",
    "# Change directory to path with all reference/input files\n",
    "%cd C:\\Users\\Brendan\\Documents\\Summer20\n",
    "\n",
    "gene = pd.read_table('gene2ensembl')\n",
    "human = pd.read_table('Homo_sapiens.gene_info')\n",
    "mart = pd.read_csv('mart_export.txt',delimiter=',')\n",
    "\n",
    "# Create Aliases column (delimited Synonyms)\n",
    "aliases = human['Synonyms'].tolist()\n",
    "aliases_sep = []\n",
    "for i in range(len(aliases)):\n",
    "    my_list = aliases[i].split(\"|\")\n",
    "    my_list.append(human['Symbol'].iloc[i])\n",
    "    aliases_sep.append(my_list)\n",
    "human['Aliases'] = aliases_sep\n",
    "\n",
    "# Create geneId column from splicing/filtering dbXrefs\n",
    "dbXrefs = human['dbXrefs'].tolist()\n",
    "geneIds = []\n",
    "for i in range(len(dbXrefs)):\n",
    "    if 'Ensembl' in dbXrefs[i]:\n",
    "        gene_id = dbXrefs[i].split(\"Ensembl:\")[1]\n",
    "    else:\n",
    "        gene_id = \"\"\n",
    "    geneIds.append(gene_id)\n",
    "human['geneId'] = geneIds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#old\n",
    "\n",
    "def translateList(list_features, testing_df):\n",
    "    \n",
    "    test_col = list_features[0]\n",
    "    reference_df = list_features[1]\n",
    "    reference_col = list_features[2]\n",
    "    output_col = list_features[3]\n",
    "\n",
    "    start_param = testing_df[test_col].tolist()\n",
    "    in_list = reference_df[reference_col].tolist()\n",
    "\n",
    "    out_listoflist = []\n",
    "\n",
    "    for i in range(len(start_param)):\n",
    "        param = start_param[i]\n",
    "\n",
    "        match_list=[]\n",
    "        for j in range(len(in_list)):\n",
    "            if str(param) in str(in_list[j]):\n",
    "                index_match = reference_df.iloc[j]\n",
    "                match_list.append(index_match)\n",
    "\n",
    "        out_list = []\n",
    "        for k in range(len(match_list)):\n",
    "            out_list.append(match_list[k][output_col])\n",
    "\n",
    "        out_listoflist.append(out_list)\n",
    "    testing_df['output '+reference_col]=out_listoflist\n",
    "\n",
    "    return testing_df\n",
    "#     'geneId (GRCH37.66)','human','geneId','Full_name_from_nomenclature_authority'\n",
    "\n",
    "def decimal_removal(database, col_index):  # make bp locs into whole numbers for translation\n",
    "    temp_list = database.iloc[:,col_index].tolist()\n",
    "    end_list = []\n",
    "    \n",
    "    for value in temp_list:\n",
    "        end_list.append(str(value).split('.',1)[0])\n",
    "        \n",
    "    return end_list\n",
    "    print(end_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def f_score(TP,FP,FN):\n",
    "    precision = TP/(TP+FP)\n",
    "    recall = TP/(TP+FN)\n",
    "    f_dia = 2 * precision * recall / (precision+recall)\n",
    "    return f_dia"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#old\n",
    "\n",
    "def scoring(dataF, colname):\n",
    "\n",
    "    nonmatches_ind = [ind for ind, x in enumerate(dataF[colname]) if len(x)==0 or x != x or x=='-']\n",
    "    matches_ind = [ind for ind, x in enumerate(dataF[colname]) if len(x)!=0 or x == x or x!='-']\n",
    "    nonmatches = dataF.iloc[nonmatches_ind]\n",
    "    matches = dataF.iloc[matches_ind]\n",
    "\n",
    " \n",
    "\n",
    "    truePos = len(dataF)-len(nonmatches)\n",
    "    falsePos = 0 # How to calculate FP?\n",
    "    falseNeg = 0\n",
    "    fScore = f_score(truePos,falsePos,falseNeg)\n",
    "#     return fScore\n",
    "    print(len(nonmatches))\n",
    "    print(len(dataF))\n",
    "    return round(100*len(nonmatches)/len(dataF),4)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "#current function\n",
    "def combined_translate(input_file, file_type, trans_list):\n",
    "    if (file_type == 'csv'):\n",
    "        df = pd.read_csv(input_file)\n",
    "    if (file_type == 'excel'):\n",
    "        df = pd.read_excel(input_file)\n",
    "    if (file_type == 'other'):\n",
    "        df = pd.read_table(input_file)\n",
    "    \n",
    "    for item in trans_list:\n",
    "        test_col = item[0]\n",
    "        reference_df = item[1]\n",
    "        reference_col = item[2]\n",
    "        output_col = item[3]\n",
    "        \n",
    "        start_param = df[test_col].tolist()\n",
    "        in_list = reference_df[reference_col].tolist()\n",
    "        out_listoflist = []\n",
    "\n",
    "        for i in range(len(start_param)):\n",
    "            param = start_param[i]\n",
    "\n",
    "            match_list=[]\n",
    "            for j in range(len(in_list)):\n",
    "                if str(param) in str(in_list[j]):\n",
    "                    index_match = reference_df.iloc[j]\n",
    "                    match_list.append(index_match)\n",
    "\n",
    "            out_list = []\n",
    "            for k in range(len(match_list)):\n",
    "                out_list.append(match_list[k][output_col])\n",
    "\n",
    "            out_listoflist.append(out_list)\n",
    "        out_colname = 'output ' + reference_col\n",
    "        df[out_colname] = out_listoflist\n",
    "\n",
    "    \n",
    "        nonmatches_ind = [ind for ind, x in enumerate(df[out_colname]) if len(x)==0 or x != x or x=='-']\n",
    "        matches_ind = [ind for ind, x in enumerate(df[out_colname]) if len(x)!=0 or x == x or x!='-']\n",
    "        nonmatches = df.iloc[nonmatches_ind]\n",
    "        matches = df.iloc[matches_ind]\n",
    "        print(round(100*len(nonmatches)/len(df),4),\" percent mismatch, \",test_col)\n",
    "    \n",
    "    print(df)\n",
    "    \n",
    "        \n",
    "          "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18\n",
      "103\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "82.5243"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4460634/\n",
    "# Supplementary materials, \n",
    "# GPEA analysis of the RNAseq UC GRN for gene families\n",
    "\n",
    "gpea = pd.read_csv('UC_GPEA.csv')\n",
    "gpea_list = ['tag',human,'Aliases','Full_name_from_nomenclature_authority']\n",
    "gpea_translated = translateList(gpea_list, gpea)\n",
    "score_gpea = 100-scoring(gpea_translated, 'output Aliases')\n",
    "# gpea_translated['output Aliases']\n",
    "score_gpea\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "' combined_translate() template\\ncopy this template and fill with parameters\\n*******   COPY   ********\\n\\n\\n\\n'"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\" combined_translate() template\n",
    "copy this template and fill with parameters\n",
    "\n",
    "*******   COPY   ********\n",
    "\n",
    "### Name of input data file\n",
    "input_file = ''\n",
    "\n",
    "### Type of file (excel, csv, or other)\n",
    "file_type = ''\n",
    "\n",
    "### translation 1 test col name\n",
    "test_col1 = ''\n",
    "\n",
    "### translation 1 reference dataset (human,gene,mart)\n",
    "ref_df1 =\n",
    "\n",
    "### translation 1 reference col name\n",
    "ref_col1 = ''\n",
    "    \n",
    "### translation 1 output col name\n",
    "out_col1 = ''\n",
    "\n",
    "### translation 1 use only if one translation, otherwise delete\n",
    "trans_list = [[test_col1,ref_df1,ref_col1,out_col1]]\n",
    "\n",
    "### translation 2 test col name (delete all until end)\n",
    "test_col2 = ''\n",
    "\n",
    "### translation 2 reference dataset (human,gene,mart)\n",
    "ref_df2 = \n",
    "\n",
    "### translation 2 reference col name\n",
    "ref_col2 = ''\n",
    "\n",
    "### translation 2 output col name\n",
    "out_col2 = ''\n",
    "\n",
    "### use for two translations\n",
    "trans_list = [[test_col1,ref_df1,ref_col1,out_col1],[test_col2,ref_df2,ref_col2,out_col2]]\n",
    "\n",
    "\n",
    "combined_translate(input_file,file_type,trans_list)\n",
    "\n",
    "************************\n",
    "\n",
    "\"\"\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        gene                                     output Aliases\n",
      "0      BMP8A                    [bone morphogenetic protein 8a]\n",
      "1   CEACAM20                    [CEA cell adhesion molecule 20]\n",
      "2       CHGA                                   [chromogranin A]\n",
      "3     COL9A1  [collagen type IX alpha 1 chain, collagen type...\n",
      "4        DCD  [desmin, isocitrate dehydrogenase (NADP(+)) 1,...\n",
      "5      GPM6A                                 [glycoprotein M6A]\n",
      "6      GRIA3  [glutamate ionotropic receptor AMPA type subun...\n",
      "7      GSTA3                [glutathione S-transferase alpha 3]\n",
      "8       IGF2  [insulin like growth factor 2, insulin like gr...\n",
      "9      KLK11                  [kallikrein related peptidase 11]\n",
      "10     KRT16  [keratin 16, keratin 16 pseudogene 4, keratin ...\n",
      "11     KRT6A                                       [keratin 6A]\n",
      "12     LEMD1  [LEM domain containing 1, LEMD1 antisense RNA ...\n",
      "13    LGALS7                          [galectin 7, galectin 7B]\n",
      "14      LY6D  [glycosylphosphatidylinositol anchored molecul...\n",
      "15      MT1H    [metallothionein 1H, metallothionein 1H like 1]\n",
      "16      PLAT  [plasminogen activator, tissue type, lysophosp...\n",
      "17      PLP1  [amyloid beta precursor like protein 1, proteo...\n",
      "18    PRSS12  [serine protease 12, transmembrane serine prot...\n",
      "19     RLBP1  [retinaldehyde binding protein 1, clavesin 2, ...\n",
      "20     SCN1A  [sodium voltage-gated channel alpha subunit 1,...\n",
      "21   TNFSF11                        [TNF superfamily member 11]\n",
      "22      UCP1                             [uncoupling protein 1]\n",
      "23    UGT2B4   [UDP glucuronosyltransferase family 2 member B4]\n",
      "Percent mismatch:  0.0\n"
     ]
    }
   ],
   "source": [
    "# https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3748065/\n",
    "# Table 2\n",
    "\n",
    "### Name of input data file\n",
    "input_file = 'TCGA.rtf'\n",
    "# input_file = 'UC_GPEA.csv'\n",
    "### Type of file (excel, csv, or other)\n",
    "file_type = 'csv'\n",
    "\n",
    "### translation 1 test col name\n",
    "test_col1 = 'gene'\n",
    "\n",
    "### translation 1 reference dataset (human,gene,mart)\n",
    "ref_df1 = human\n",
    "\n",
    "### translation 1 reference col name\n",
    "ref_col1 = 'Aliases'\n",
    "    \n",
    "### translation 1 output col name\n",
    "out_col1 = 'Full_name_from_nomenclature_authority'\n",
    "\n",
    "### translation 1 use only if one translation, otherwise delete\n",
    "trans_list1 = [[test_col1,ref_df1,ref_col1,out_col1]]\n",
    "\n",
    "\n",
    "combined_translate(input_file,file_type,trans_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17.4757  percent mismatch,  tag\n",
      "          tag                                               name  \\\n",
      "0          CD                                       CD molecules   \n",
      "1        HIST                   Histones / Replication-dependent   \n",
      "2       KRTAP                        Keratin associated proteins   \n",
      "3       SNORA  ncRNAs / Small nucleolar RNAs : H/ACA box cont...   \n",
      "4       ZKRAB                                                  -   \n",
      "5         ZNF                             Zinc fingers C2H2-type   \n",
      "6       PCDHC             Cadherins / Protocadherins : Clustered   \n",
      "7         RPL                               L ribosomal proteins   \n",
      "8         KRT                                                  -   \n",
      "9         HLA                         Histocompatibility complex   \n",
      "10        IGD  Immunoglobulin superfamily / Immunoglobulin-li...   \n",
      "11     LNCRNA                               Long non-coding RNAs   \n",
      "12      C1SET  Immunoglobulin superfamily / C1-set domain con...   \n",
      "13       HOXL            Homeoboxes / ANTP class : HOXL subclass   \n",
      "14        LCE                           Late cornified envelopes   \n",
      "15        RPS                               S ribosomal proteins   \n",
      "16        OR2               GPCR / Class A : Olfactory receptors   \n",
      "17       IFF1    Intermediate filaments type I keratins (acidic)   \n",
      "18       VSET  Immunoglobulin superfamily / V-set domain cont...   \n",
      "19      PRAME                                                  -   \n",
      "20        KLK                                        Kallikreins   \n",
      "21        RBM                 RNA binding motif (RRM) containing   \n",
      "22   COLLAGEN                                          Collagens   \n",
      "23        KIR          Killer cell immunoglobulin-like receptors   \n",
      "24        CYP                                   Cytochrome P450s   \n",
      "25       IFF2    Intermediate filaments type II keratins (basic)   \n",
      "26      ZSCAN                                                  -   \n",
      "27        PSM            Proteasome (prosome macropain) subunits   \n",
      "28       PAR1                     Pseudoautosomal regions / PAR1   \n",
      "29     SERPIN          Serine (or cysteine) peptidase inhibitors   \n",
      "..        ...                                                ...   \n",
      "73       OR10               GPCR / Class A : Olfactory receptors   \n",
      "74        NKL             Homeoboxes / ANTP class : NKL subclass   \n",
      "75       BPIF                                BPI fold containing   \n",
      "76     EFHAND                          EF-hand domain containing   \n",
      "77        TUB                                           Tubulins   \n",
      "78       SH2D                              SH2 domain containing   \n",
      "79       ACOT                             Acyl CoA thioesterases   \n",
      "80      MYHII            Myosins / Myosin superfamily : Class II   \n",
      "81       OR56               GPCR / Class A : Olfactory receptors   \n",
      "82        ITG                                          Integrins   \n",
      "83       MCDH                        Cadherins / Major cadherins   \n",
      "84       BTBD                          BTB/POZ domain containing   \n",
      "85     SIGLEC                Sialic acid binding Ig-like lectins   \n",
      "86     TMPRSS                  Serine peptidases / Transmembrane   \n",
      "87        GGT                         Gamma-glutamyltransferases   \n",
      "88       bHLH                    Basic helix-loop-helix proteins   \n",
      "89       CACN                           Calcium channel subunits   \n",
      "90      SDRC1  Short chain dehydrogenase/reductase superfamil...   \n",
      "91        FN3             Fibronectin type III domain containing   \n",
      "92       SIRP                         Signal-regulatory proteins   \n",
      "93        TTC   Tetratricopeptide (TTC) repeat domain containing   \n",
      "94        IFT                  Intraflagellar transport homologs   \n",
      "95        SIX                            Homeoboxes / SINE class   \n",
      "96       ZMYM                              Zinc fingers MYM type   \n",
      "97       DEFB                                     Defensins beta   \n",
      "98       FIBC                     Fibrinogen C domain containing   \n",
      "99     ADAMTS  ADAM metallopeptidases with thrombospondin typ...   \n",
      "100       OR6               GPCR / Class A : Olfactory receptors   \n",
      "101       AQP                          Ion channels / Aquaporins   \n",
      "102     LGALS                        Lectins galactoside-binding   \n",
      "\n",
      "                                        output Aliases  \n",
      "0    [acyl-CoA dehydrogenase very long chain, ADAM ...  \n",
      "1    [H1.2 linker histone, cluster member, H1.3 lin...  \n",
      "2    [keratin associated protein 5-9, keratin assoc...  \n",
      "3    [small nucleolar RNA, H/ACA box 63, small nucl...  \n",
      "4                                                   []  \n",
      "5    [adrenoceptor alpha 2A, zinc finger homeobox 3...  \n",
      "6                                                   []  \n",
      "7    [poly(ADP-ribose) polymerase family member 4, ...  \n",
      "8    [keratin associated protein 5-9, keratin 1, ke...  \n",
      "9    [CD74 molecule, G protein subunit alpha o1, ho...  \n",
      "10   [DCC netrin 1 receptor, forkhead box C1, immun...  \n",
      "11   [testis-specific transcript, Y-linked 3B, C5 3...  \n",
      "12                                                  []  \n",
      "13                                                  []  \n",
      "14   [hydroxyacyl-CoA dehydrogenase trifunctional m...  \n",
      "15   [exostosin glycosyltransferase 1, FAU ubiquiti...  \n",
      "16   [immunoglobulin kappa variable 3/OR2-268 (non-...  \n",
      "17                                                  []  \n",
      "18                                                  []  \n",
      "19   [preferentially expressed antigen in melanoma,...  \n",
      "20   [kallikrein related peptidase 3, kallikrein 1,...  \n",
      "21   [four and a half LIM domains 1, RBMX pseudogen...  \n",
      "22                                                  []  \n",
      "23   [GTP binding protein overexpressed in skeletal...  \n",
      "24   [acylphosphatase 1, acylphosphatase 2, cytochr...  \n",
      "25                                                  []  \n",
      "26   [paternally expressed 3, zinc finger protein 1...  \n",
      "27   [folate hydrolase 1, inositol polyphosphate ph...  \n",
      "28   [lysophosphatidic acid receptor 1, coagulation...  \n",
      "29   [serpin family A member 3, angiotensinogen, se...  \n",
      "..                                                 ...  \n",
      "73   [olfactory receptor family 10 subfamily J memb...  \n",
      "74   [TNF superfamily member 11, ankyrin repeat and...  \n",
      "75   [bactericidal permeability increasing protein,...  \n",
      "76                                                  []  \n",
      "77   [TUB bipartite transcription factor, tubulin a...  \n",
      "78   [SH2 domain containing 1A, BCAR3 adaptor prote...  \n",
      "79   [acyl-CoA thioesterase 8, acyl-CoA thioesteras...  \n",
      "80                                                  []  \n",
      "81   [olfactory receptor family 56 subfamily A memb...  \n",
      "82   [integrin subunit alpha 6, integrin subunit al...  \n",
      "83                                                  []  \n",
      "84   [BTB domain and CNC homolog 1, calicin, galect...  \n",
      "85   [CD22 molecule, CD33 molecule, sialic acid bin...  \n",
      "86   [hepsin, transmembrane serine protease 15, sup...  \n",
      "87   [CHM Rab escort protein, farnesyltransferase, ...  \n",
      "88   [aryl hydrocarbon receptor, aryl hydrocarbon r...  \n",
      "89   [calcium voltage-gated channel subunit alpha1 ...  \n",
      "90                                                  []  \n",
      "91   [POU class 3 homeobox 4, schlafen family membe...  \n",
      "92   [signal regulatory protein beta 1, signal regu...  \n",
      "93   [chaperonin containing TCP1 subunit 6A, mitoch...  \n",
      "94   [intraflagellar transport 88, intraflagellar t...  \n",
      "95   [SIX homeobox 6, SIX homeobox 1, SIX homeobox ...  \n",
      "96   [zinc finger MYM-type containing 2, zinc finge...  \n",
      "97   [defensin beta 1, defensin beta 4A, defensin b...  \n",
      "98                  [fibrinogen C domain containing 1]  \n",
      "99   [ADAM metallopeptidase with thrombospondin typ...  \n",
      "100  [olfactory receptor family 6 subfamily A membe...  \n",
      "101  [aquaporin 8, aquaporin 1 (Colton blood group)...  \n",
      "102  [Charcot-Leyden crystal galectin, galectin 1, ...  \n",
      "\n",
      "[103 rows x 3 columns]\n"
     ]
    }
   ],
   "source": [
    "# https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4460634/\n",
    "# Supplementary materials, \n",
    "# GPEA analysis of the RNAseq UC GRN for gene families\n",
    "\n",
    "### Name of input data file\n",
    "input_file = 'UC_GPEA.csv'\n",
    "\n",
    "### Type of file (excel, csv, or other)\n",
    "file_type = 'csv'\n",
    "\n",
    "### translation 1 test col name\n",
    "test_col1 = 'tag'\n",
    "\n",
    "### translation 1 reference dataset (human,gene,mart)\n",
    "ref_df1 = human\n",
    "\n",
    "### translation 1 reference col name\n",
    "ref_col1 = 'Aliases'\n",
    "    \n",
    "### translation 1 output col name\n",
    "out_col1 = 'Full_name_from_nomenclature_authority'\n",
    "\n",
    "### translation 1 use only if one translation, otherwise delete\n",
    "trans_list = [[test_col1,ref_df1,ref_col1,out_col1]]\n",
    "\n",
    "combined_translate(input_file,file_type,trans_list)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10.7143  percent mismatch,  Gene\n",
      "100.0  percent mismatch,  Protein\n",
      "            Gene                                        Protein  \\\n",
      "0          OPRM1                              μ-opioid receptor   \n",
      "1          OPRK1                              κ-opioid receptor   \n",
      "2           PDYN                                Preprodynorphin   \n",
      "3             TH                           Tyrosine hydroxylase   \n",
      "4           DRD2                           Dopamine receptor D2   \n",
      "5           DRD3                           Dopamine receptor D3   \n",
      "6           DRD4                           Dopamine receptor D4   \n",
      "7            DBH                        Dopamine β-hydroxylase    \n",
      "8   DAT (SLC6A3)                           Dopamine transporter   \n",
      "9           TPH1                       Tryptophan hydroxylase 1   \n",
      "10          TPH2                       Tryptophan hydroxylase 2   \n",
      "11         HTR1B                          Serotonin receptor 1B   \n",
      "12         HTR2A                          Serotonin receptor 2A   \n",
      "13  SERT (SLCA4)                          Serotonin transporter   \n",
      "14          MAOA                            Monoamine oxidase A   \n",
      "15          COMT                   Catechol-O-methyltransferase   \n",
      "16        GABRA1                      GABA receptor subunit α-1   \n",
      "17        GABRA6                      GABA receptor subunit α-6   \n",
      "18        GABRB1                      GABA receptor subunit β-1   \n",
      "19         CHRM2           Muscarinic acetylcholine receptor M2   \n",
      "20          CNR1                         Cannabinoid receptor 1   \n",
      "21          FAAH                     Fatty acid amide hydrolase   \n",
      "22           NPY                                 Neuropeptide Y   \n",
      "23         ADH1B                       Alcohol dehydrogenase 1B   \n",
      "24         ADH1C                       Alcohol dehydrogenase 1C   \n",
      "25         ALDH2                       Aldehyde dehydrogenase 2   \n",
      "26        CYP2d^                              Cytochrome CYP450   \n",
      "27         ANKK1  Ankyrin repeat and kinase domain-containing 1   \n",
      "\n",
      "   Chromosomal location                                     output Aliases  \\\n",
      "0              6q24-q25                             [opioid receptor mu 1]   \n",
      "1                8q11.2                          [opioid receptor kappa 1]   \n",
      "2          20pter-p12.2               [prodynorphin, PDYN antisense RNA 1]   \n",
      "3               11p15.5  [acetyl-CoA acyltransferase 1, acetyl-CoA acet...   \n",
      "4                 11q23  [dopamine receptor D2, SET domain bifurcated h...   \n",
      "5                3q13.3  [dopamine receptor D3, tumor protein p53 bindi...   \n",
      "6               11p15.5     [dopamine receptor D4, ring finger protein 17]   \n",
      "7                  9q34   [dopamine beta-hydroxylase, DBH antisense RNA 1]   \n",
      "8                5p15.3                                                 []   \n",
      "9           11p15.3-p14  [protein tyrosine phosphatase non-receptor typ...   \n",
      "10              12q21.1  [tryptophan 2,3-dioxygenase, tryptophan hydrox...   \n",
      "11                 6q13                  [5-hydroxytryptamine receptor 1B]   \n",
      "12            13q14-q21  [5-hydroxytryptamine receptor 2A, HTR2A antise...   \n",
      "13          16q11.1-q12                                                 []   \n",
      "14              Xp11.23                              [monoamine oxidase A]   \n",
      "15              22q11.2  [catechol-O-methyltransferase, catechol-O-meth...   \n",
      "16             5q34-q35  [gamma-aminobutyric acid type A receptor subun...   \n",
      "17           5q31.1-q35  [gamma-aminobutyric acid type A receptor subun...   \n",
      "18             4p13-p12  [gamma-aminobutyric acid type A receptor subun...   \n",
      "19             7q35-q36                [cholinergic receptor muscarinic 2]   \n",
      "20             6q14-q15  [cannabinoid receptor 1, protocadherin alpha c...   \n",
      "21              1p35-34  [fatty acid amide hydrolase, fatty acid 2-hydr...   \n",
      "22               7p15.1  [neuropeptide Y, neuropeptide Y receptor Y1, n...   \n",
      "23                 4q22  [alcohol dehydrogenase 1B (class I), beta poly...   \n",
      "24                 4q22  [alcohol dehydrogenase 1C (class I), gamma pol...   \n",
      "25              12q24.2  [aldehyde dehydrogenase 2 family member, aldeh...   \n",
      "26              22q13.1                                                 []   \n",
      "27              11q23.2    [ankyrin repeat and kinase domain containing 1]   \n",
      "\n",
      "   output description  \n",
      "0                  []  \n",
      "1                  []  \n",
      "2                  []  \n",
      "3                  []  \n",
      "4                  []  \n",
      "5                  []  \n",
      "6                  []  \n",
      "7                  []  \n",
      "8                  []  \n",
      "9                  []  \n",
      "10                 []  \n",
      "11                 []  \n",
      "12                 []  \n",
      "13                 []  \n",
      "14                 []  \n",
      "15                 []  \n",
      "16                 []  \n",
      "17                 []  \n",
      "18                 []  \n",
      "19                 []  \n",
      "20                 []  \n",
      "21                 []  \n",
      "22                 []  \n",
      "23                 []  \n",
      "24                 []  \n",
      "25                 []  \n",
      "26                 []  \n",
      "27                 []  \n"
     ]
    }
   ],
   "source": [
    "# Drug study from google docs file\n",
    "\n",
    "### Name of input data file\n",
    "input_file = 'drug_supplement.xlsx'\n",
    "\n",
    "### Type of file (excel, csv, or other)\n",
    "file_type = 'excel'\n",
    "\n",
    "### translation 1 test col name\n",
    "test_col1 = 'Gene'\n",
    "\n",
    "### translation 1 reference dataset (human,gene,mart)\n",
    "ref_df1 = human\n",
    "\n",
    "### translation 1 reference col name\n",
    "ref_col1 = 'Aliases'\n",
    "    \n",
    "### translation 1 output col name\n",
    "out_col1 = 'description'\n",
    "\n",
    "### translation 2 test col name (delete all until end)\n",
    "test_col2 = 'Protein'\n",
    "\n",
    "### translation 2 reference dataset (human,gene,mart)\n",
    "ref_df2 = human\n",
    "\n",
    "### translation 2 reference col name\n",
    "ref_col2 = 'description'\n",
    "\n",
    "### translation 2 output col name\n",
    "out_col2 = 'Symbol'\n",
    "\n",
    "### use for two translations\n",
    "trans_list = [[test_col1,ref_df1,ref_col1,out_col1],[test_col2,ref_df2,ref_col2,out_col2]]\n",
    "\n",
    "\n",
    "combined_translate(input_file,file_type,trans_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
